<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Analysis of the fibrosis datataset • condimentsPaper</title>

<!-- favicons -->
<link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png" />
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png" />
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png" />
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png" />

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous" />


<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Analysis of the fibrosis datataset" />
<meta property="og:description" content="condimentsPaper" />
<meta property="og:image" content="http://hectorrdb.github.io//condimentsPaper/logo.png" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">condimentsPaper</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/Examples.html">Figure 1: examples</a>
    </li>
    <li>
      <a href="../articles/Fibrosis.html">Analysis of the fibrosis datataset</a>
    </li>
    <li>
      <a href="../articles/KRAS.html">Analysis of the KRAS datataset</a>
    </li>
    <li>
      <a href="../articles/Simulations.html">Running and analyzing the simulations</a>
    </li>
    <li>
      <a href="../articles/TCDD.html">Analysis of the TCDD datataset</a>
    </li>
    <li>
      <a href="../articles/TGFB.html">Analysis of the TGFB datataset</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/HectorRDB/condimentsPaper">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<script src="Fibrosis_files/header-attrs-2.9/header-attrs.js"></script>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Analysis of the fibrosis datataset</h1>
                        <h4 class="author">Hector Roux de Bézieux</h4>
            
      
      
      <div class="hidden name"><code>Fibrosis.Rmd</code></div>

    </div>

    
    
<div id="load-data" class="section level1">
<h1>Load data</h1>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a>libs <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;here&quot;</span>, <span class="st">&quot;tidyr&quot;</span>, <span class="st">&quot;dplyr&quot;</span>, <span class="st">&quot;Seurat&quot;</span>, <span class="st">&quot;scater&quot;</span>, <span class="st">&quot;slingshot&quot;</span>, <span class="st">&quot;cowplot&quot;</span>,</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>          <span class="st">&quot;condiments&quot;</span>, <span class="st">&quot;ggplot2&quot;</span>, <span class="st">&quot;readr&quot;</span>, <span class="st">&quot;tradeSeq&quot;</span>,  <span class="st">&quot;pheatmap&quot;</span>, <span class="st">&quot;fgsea&quot;</span>,</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a>          <span class="st">&quot;msigdbr&quot;</span>, <span class="st">&quot;scales&quot;</span>)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">suppressMessages</span>(</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">suppressWarnings</span>(<span class="fu">sapply</span>(libs, require, <span class="at">character.only =</span> <span class="cn">TRUE</span>))</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>
<pre><code>##       here      tidyr      dplyr     Seurat     scater  slingshot    cowplot 
##       TRUE       TRUE       TRUE       TRUE       TRUE       TRUE       TRUE 
## condiments    ggplot2      readr   tradeSeq   pheatmap      fgsea    msigdbr 
##       TRUE       TRUE       TRUE       TRUE       TRUE       TRUE       TRUE 
##     scales 
##       TRUE</code></pre>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(libs)</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="fu">theme_set</span>(<span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>            <span class="fu">theme</span>(<span class="at">rect =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>                  <span class="at">plot.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>                  <span class="at">strip.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="st">&quot;transparent&quot;</span>)</span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>                  ))</span></code></pre></div>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>fibrosis <span class="ot">&lt;-</span> condimentsPaper<span class="sc">::</span><span class="fu">import_fibrosis</span>()</span></code></pre></div>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;fibrosis&quot;</span>, <span class="at">package =</span> <span class="st">&quot;condimentsPaper&quot;</span>)</span></code></pre></div>
</div>
<div id="eda" class="section level1">
<h1>EDA</h1>
<p>Reduced dimension coordinates are obtained from the original publication.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">bind_cols</span>(<span class="fu">reducedDim</span>(fibrosis, <span class="st">&quot;UMAP&quot;</span>) <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>(),</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>                <span class="fu">colData</span>(fibrosis)[, <span class="dv">1</span><span class="sc">:</span><span class="dv">14</span>] <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>())</span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(df, <span class="fu">aes</span>(<span class="at">x =</span> UMAP_1, <span class="at">y =</span> UMAP_2, <span class="at">col =</span> Status)) <span class="sc">+</span></span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> .<span class="dv">5</span>) <span class="sc">+</span></span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_brewer</span>(<span class="at">palette =</span> <span class="st">&quot;Accent&quot;</span>) <span class="sc">+</span></span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">col =</span> <span class="st">&quot;Disease&quot;</span>) <span class="sc">+</span></span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="fu">c</span>(.<span class="dv">15</span>, .<span class="dv">4</span>),</span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.background =</span> <span class="fu">element_blank</span>()) <span class="sc">+</span></span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">col =</span> <span class="fu">guide_legend</span>(<span class="at">override.aes =</span> <span class="fu">list</span>(<span class="at">size =</span> <span class="dv">3</span>)))</span></code></pre></div>
<p><img src="Fibrosis_files/figure-html/unnamed-chunk-4-1.png" width="700"  style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(df, <span class="fu">aes</span>(<span class="at">x =</span> UMAP_1, <span class="at">y =</span> UMAP_2, <span class="at">col =</span> celltype)) <span class="sc">+</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> .<span class="dv">5</span>) <span class="sc">+</span></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_brewer</span>(<span class="at">palette =</span> <span class="st">&quot;Dark2&quot;</span>) <span class="sc">+</span></span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">col =</span> <span class="st">&quot;Cell Type&quot;</span>) <span class="sc">+</span></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="fu">c</span>(.<span class="dv">2</span>, .<span class="dv">4</span>),</span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.background =</span> <span class="fu">element_blank</span>()) <span class="sc">+</span></span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">col =</span> <span class="fu">guide_legend</span>(<span class="at">override.aes =</span> <span class="fu">list</span>(<span class="at">size =</span> <span class="dv">3</span>)))</span></code></pre></div>
<p><img src="Fibrosis_files/figure-html/unnamed-chunk-4-2.png" width="700"  style="display: block; margin: auto;" /></p>
</div>
<div id="imbalance-score" class="section level1">
<h1>Imbalance score</h1>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>scores <span class="ot">&lt;-</span> condiments<span class="sc">::</span><span class="fu">imbalance_score</span>(</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">Object =</span> df <span class="sc">%&gt;%</span> <span class="fu">select</span>(UMAP_1, UMAP_2) <span class="sc">%&gt;%</span> <span class="fu">as.matrix</span>(), </span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">conditions =</span> df<span class="sc">$</span>Status, <span class="at">k =</span> <span class="dv">20</span>, <span class="at">smooth =</span> <span class="dv">30</span>)</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>df<span class="sc">$</span>scores <span class="ot">&lt;-</span> scores<span class="sc">$</span>scaled_scores</span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(df, <span class="fu">aes</span>(<span class="at">x =</span> UMAP_1, <span class="at">y =</span> UMAP_2, <span class="at">col =</span> scores)) <span class="sc">+</span></span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> .<span class="dv">7</span>) <span class="sc">+</span></span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_viridis_c</span>(<span class="at">option =</span> <span class="st">&quot;C&quot;</span>, <span class="at">breaks =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">2</span>, <span class="dv">4</span>, <span class="dv">6</span>)) <span class="sc">+</span></span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">col =</span> <span class="st">&quot;Scores&quot;</span>) <span class="sc">+</span></span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="fu">c</span>(.<span class="dv">15</span>, .<span class="dv">4</span>),</span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.background =</span> <span class="fu">element_blank</span>())</span></code></pre></div>
<p><img src="Fibrosis_files/figure-html/unnamed-chunk-5-1.png" width="700"  style="display: block; margin: auto;" /></p>
</div>
<div id="differential-topology" class="section level1">
<h1>Differential Topology</h1>
<p>To estimate the trajectory, we use <em>slingshot</em> <span class="citation">(Street et al. 2018)</span>.</p>
<div id="fit-slingshot" class="section level2">
<h2>Fit slingshot</h2>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>fibrosis <span class="ot">&lt;-</span> <span class="fu">slingshot</span>(fibrosis, <span class="at">reducedDim =</span> <span class="st">&#39;UMAP&#39;</span>,</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>                 <span class="at">clusterLabels =</span> <span class="fu">colData</span>(fibrosis)<span class="sc">$</span>celltype,</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>                 <span class="at">start.clus =</span> <span class="st">&#39;AT1&#39;</span>, <span class="at">approx_points =</span> <span class="dv">100</span>)</span></code></pre></div>
</div>
<div id="topology-test" class="section level2">
<h2>Topology Test</h2>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">821</span>)</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a><span class="fu">topologyTest</span>(<span class="fu">SlingshotDataSet</span>(fibrosis), fibrosis<span class="sc">$</span>Status)</span></code></pre></div>
<pre><code>## Generating permuted trajectories</code></pre>
<pre><code>## Running KS-mean test</code></pre>
<pre><code>##    method thresh  statistic      p.value
## 1 KS_mean   0.01 0.04183036 2.964406e-12</code></pre>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>fibrosis_SCGB3A2 <span class="ot">&lt;-</span> <span class="fu">slingshot</span>(fibrosis[, fibrosis<span class="sc">$</span>celltype <span class="sc">!=</span> <span class="st">&quot;SCGB3A2+&quot;</span>],</span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>                         <span class="at">reducedDim =</span> <span class="st">&#39;UMAP&#39;</span>,</span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>            <span class="at">clusterLabels =</span> <span class="fu">colData</span>(fibrosis[, fibrosis<span class="sc">$</span>celltype <span class="sc">!=</span> <span class="st">&quot;SCGB3A2+&quot;</span>])<span class="sc">$</span>celltype,</span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>                 <span class="at">start.clus =</span> <span class="st">&#39;AT1&#39;</span>, <span class="at">approx_points =</span> <span class="dv">100</span>)</span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">821</span>)</span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a><span class="fu">topologyTest</span>(<span class="fu">SlingshotDataSet</span>(fibrosis_SCGB3A2), fibrosis_SCGB3A2<span class="sc">$</span>Status)</span></code></pre></div>
<pre><code>## Generating permuted trajectories
## Running KS-mean test</code></pre>
<pre><code>##    method thresh   statistic p.value
## 1 KS_mean   0.01 0.002720157       1</code></pre>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(fibrosis_SCGB3A2)</span></code></pre></div>
</div>
<div id="separate-trajectories" class="section level2">
<h2>Separate trajectories</h2>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a>sdss <span class="ot">&lt;-</span> <span class="fu">slingshot_conditions</span>(<span class="fu">SlingshotDataSet</span>(fibrosis), fibrosis<span class="sc">$</span>Status)</span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>curves <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(<span class="fu">lapply</span>(sdss, slingCurves, <span class="at">as.df =</span> <span class="cn">TRUE</span>),</span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">.id =</span> <span class="st">&quot;Status&quot;</span>)</span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a>p4a <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(df, <span class="fu">aes</span>(<span class="at">x =</span> UMAP_1, <span class="at">y =</span> UMAP_2, <span class="at">col =</span> Status)) <span class="sc">+</span></span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> .<span class="dv">7</span>, <span class="at">alpha =</span> .<span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_brewer</span>(<span class="at">palette =</span> <span class="st">&quot;Accent&quot;</span>) <span class="sc">+</span></span>
<span id="cb18-8"><a href="#cb18-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">col =</span> <span class="st">&quot;Disaes&quot;</span>) <span class="sc">+</span></span>
<span id="cb18-9"><a href="#cb18-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_path</span>(<span class="at">data =</span> curves <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(Status, Lineage, Order),</span>
<span id="cb18-10"><a href="#cb18-10" aria-hidden="true" tabindex="-1"></a>            <span class="fu">aes</span>(<span class="at">group =</span> <span class="fu">interaction</span>(Lineage, Status)), <span class="at">size =</span> <span class="fl">1.5</span>) <span class="sc">+</span></span>
<span id="cb18-11"><a href="#cb18-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">&quot;text&quot;</span>, <span class="at">x =</span> <span class="sc">-</span><span class="dv">10</span>, <span class="at">y =</span> <span class="dv">6</span>, <span class="at">label =</span> <span class="st">&quot;Lineage1&quot;</span>, <span class="at">size =</span> <span class="dv">5</span>) <span class="sc">+</span></span>
<span id="cb18-12"><a href="#cb18-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">&quot;text&quot;</span>, <span class="at">x =</span> <span class="sc">-</span><span class="dv">7</span>, <span class="at">y =</span> <span class="sc">-</span><span class="fl">2.7</span>, <span class="at">label =</span> <span class="st">&quot;Lineage2&quot;</span>, <span class="at">size =</span> <span class="dv">5</span>) <span class="sc">+</span></span>
<span id="cb18-13"><a href="#cb18-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="fu">c</span>(.<span class="dv">15</span>, .<span class="dv">35</span>),</span>
<span id="cb18-14"><a href="#cb18-14" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.background =</span> <span class="fu">element_blank</span>()) <span class="sc">+</span></span>
<span id="cb18-15"><a href="#cb18-15" aria-hidden="true" tabindex="-1"></a>  <span class="cn">NULL</span></span>
<span id="cb18-16"><a href="#cb18-16" aria-hidden="true" tabindex="-1"></a>p4a</span></code></pre></div>
<p><img src="Fibrosis_files/figure-html/unnamed-chunk-8-1.png" width="700"  style="display: block; margin: auto;" /></p>
</div>
<div id="common-trajectory" class="section level2">
<h2>Common trajectory</h2>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">bind_cols</span>(</span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>(<span class="fu">reducedDim</span>(fibrosis, <span class="st">&quot;UMAP&quot;</span>)),</span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slingPseudotime</span>(fibrosis) <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">rename_with</span>(paste0, <span class="st">&quot;_pst&quot;</span>, <span class="at">.cols =</span> <span class="fu">everything</span>()),</span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slingCurveWeights</span>(fibrosis) <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>(),</span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Lineage1_pst =</span> <span class="fu">if_else</span>(<span class="fu">is.na</span>(Lineage1_pst), <span class="dv">0</span>, Lineage1_pst),</span>
<span id="cb19-8"><a href="#cb19-8" aria-hidden="true" tabindex="-1"></a>         <span class="at">Lineage2_pst =</span> <span class="fu">if_else</span>(<span class="fu">is.na</span>(Lineage2_pst), <span class="dv">0</span>, Lineage2_pst),</span>
<span id="cb19-9"><a href="#cb19-9" aria-hidden="true" tabindex="-1"></a>         <span class="at">pst =</span> <span class="fu">if_else</span>(Lineage1 <span class="sc">&gt;</span> Lineage2, Lineage1_pst, Lineage2_pst),</span>
<span id="cb19-10"><a href="#cb19-10" aria-hidden="true" tabindex="-1"></a>         <span class="at">pst =</span> <span class="fu">max</span>(pst) <span class="sc">-</span> pst)</span>
<span id="cb19-11"><a href="#cb19-11" aria-hidden="true" tabindex="-1"></a>curves <span class="ot">&lt;-</span> <span class="fu">slingCurves</span>(fibrosis, <span class="at">as.df =</span> <span class="cn">TRUE</span>)</span>
<span id="cb19-12"><a href="#cb19-12" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(df, <span class="fu">aes</span>(<span class="at">x =</span> UMAP_1, <span class="at">y =</span> UMAP_2)) <span class="sc">+</span></span>
<span id="cb19-13"><a href="#cb19-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> .<span class="dv">7</span>, <span class="fu">aes</span>(<span class="at">col =</span> pst)) <span class="sc">+</span></span>
<span id="cb19-14"><a href="#cb19-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_viridis_c</span>() <span class="sc">+</span></span>
<span id="cb19-15"><a href="#cb19-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">col =</span> <span class="st">&quot;Pseudotime&quot;</span>) <span class="sc">+</span></span>
<span id="cb19-16"><a href="#cb19-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_path</span>(<span class="at">data =</span> curves <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(Order),</span>
<span id="cb19-17"><a href="#cb19-17" aria-hidden="true" tabindex="-1"></a>            <span class="fu">aes</span>(<span class="at">group =</span> Lineage), <span class="at">col =</span> <span class="st">&quot;black&quot;</span>, <span class="at">size =</span> <span class="fl">1.5</span>) <span class="sc">+</span></span>
<span id="cb19-18"><a href="#cb19-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">&quot;text&quot;</span>, <span class="at">x =</span> <span class="sc">-</span><span class="dv">10</span>, <span class="at">y =</span> <span class="dv">6</span>, <span class="at">label =</span> <span class="st">&quot;Lineage1&quot;</span>, <span class="at">size =</span> <span class="dv">5</span>) <span class="sc">+</span></span>
<span id="cb19-19"><a href="#cb19-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">&quot;text&quot;</span>, <span class="at">x =</span> <span class="sc">-</span><span class="dv">7</span>, <span class="at">y =</span> <span class="sc">-</span><span class="fl">2.7</span>, <span class="at">label =</span> <span class="st">&quot;Lineage2&quot;</span>, <span class="at">size =</span> <span class="dv">5</span>) <span class="sc">+</span></span>
<span id="cb19-20"><a href="#cb19-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="fu">c</span>(.<span class="dv">15</span>, .<span class="dv">35</span>),</span>
<span id="cb19-21"><a href="#cb19-21" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.background =</span> <span class="fu">element_blank</span>()) </span></code></pre></div>
<p><img src="Fibrosis_files/figure-html/unnamed-chunk-9-1.png" width="700"  style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="differential-progression" class="section level1">
<h1>Differential Progression</h1>
<div id="test" class="section level2">
<h2>Test</h2>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="fu">progressionTest</span>(fibrosis, <span class="at">conditions =</span> fibrosis<span class="sc">$</span>Status, <span class="at">lineages =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<pre><code>##   lineage  statistic     p.value
## 1     All 10.6182770 1.22521e-26
## 2       1  0.6388885 2.20000e-16
## 3       2  0.1998837 2.20000e-16</code></pre>
</div>
<div id="plot" class="section level2">
<h2>Plot</h2>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span>  <span class="fu">slingPseudotime</span>(fibrosis) <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>() </span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>df<span class="sc">$</span>Status <span class="ot">&lt;-</span> fibrosis<span class="sc">$</span>Status</span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> df <span class="sc">%&gt;%</span> </span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>Status, <span class="at">names_to =</span> <span class="st">&quot;Lineage&quot;</span>,</span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a>               <span class="at">values_to =</span> <span class="st">&quot;pst&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(pst))</span>
<span id="cb22-7"><a href="#cb22-7" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(df, <span class="fu">aes</span>(<span class="at">x =</span> pst)) <span class="sc">+</span></span>
<span id="cb22-8"><a href="#cb22-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_density</span>(<span class="at">alpha =</span> .<span class="dv">8</span>, <span class="fu">aes</span>(<span class="at">fill =</span> Status), <span class="at">col =</span> <span class="st">&quot;transparent&quot;</span>) <span class="sc">+</span></span>
<span id="cb22-9"><a href="#cb22-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_density</span>(<span class="fu">aes</span>(<span class="at">col =</span> Status), <span class="at">fill =</span> <span class="st">&quot;transparent&quot;</span>, <span class="at">size =</span> <span class="fl">1.5</span>) <span class="sc">+</span></span>
<span id="cb22-10"><a href="#cb22-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;Pseudotime&quot;</span>, <span class="at">fill =</span> <span class="st">&quot;Status&quot;</span>) <span class="sc">+</span></span>
<span id="cb22-11"><a href="#cb22-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>Lineage, <span class="at">scales =</span> <span class="st">&quot;free_x&quot;</span>) <span class="sc">+</span></span>
<span id="cb22-12"><a href="#cb22-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">col =</span> <span class="st">&quot;none&quot;</span>, <span class="at">fill =</span> <span class="fu">guide_legend</span>(</span>
<span id="cb22-13"><a href="#cb22-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">override.aes =</span> <span class="fu">list</span>(<span class="at">size =</span> <span class="fl">1.5</span>, <span class="at">col =</span> <span class="fu">c</span>(<span class="st">&quot;#7FC97F&quot;</span>, <span class="st">&quot;#BEAED4&quot;</span>))</span>
<span id="cb22-14"><a href="#cb22-14" aria-hidden="true" tabindex="-1"></a>  )) <span class="sc">+</span></span>
<span id="cb22-15"><a href="#cb22-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_brewer</span>(<span class="at">palette =</span> <span class="st">&quot;Accent&quot;</span>) <span class="sc">+</span></span>
<span id="cb22-16"><a href="#cb22-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_brewer</span>(<span class="at">palette =</span> <span class="st">&quot;Accent&quot;</span>) <span class="sc">+</span></span>
<span id="cb22-17"><a href="#cb22-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="fu">c</span>(.<span class="dv">8</span>, .<span class="dv">8</span>), <span class="at">legend.direction =</span> <span class="st">&quot;horizontal&quot;</span>)</span></code></pre></div>
<p><img src="Fibrosis_files/figure-html/unnamed-chunk-11-1.png" width="700"  style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="differential-fate-selection" class="section level1">
<h1>Differential fate selection</h1>
<div id="test-1" class="section level2">
<h2>Test</h2>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="fu">fateSelectionTest</span>(fibrosis, <span class="at">conditions =</span> fibrosis<span class="sc">$</span>Status)</span></code></pre></div>
<pre><code>## note: only 1 unique complexity parameters in default grid. Truncating the grid to 1 .</code></pre>
<pre><code>## Loading required package: lattice</code></pre>
<pre><code>## note: only 1 unique complexity parameters in default grid. Truncating the grid to 1 .</code></pre>
<pre><code>## Registered S3 method overwritten by &#39;cli&#39;:
##   method     from         
##   print.boxx spatstat.geom</code></pre>
<pre><code>## # A tibble: 1 × 3
##   pair  statistic  p.value
##   &lt;chr&gt;     &lt;dbl&gt;    &lt;dbl&gt;
## 1 1vs2      0.647 7.95e-64</code></pre>
</div>
<div id="plot-1" class="section level2">
<h2>Plot</h2>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">slingCurveWeights</span>(<span class="fu">SlingshotDataSet</span>(fibrosis), <span class="at">as.probs =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>()</span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a>df<span class="sc">$</span>Status <span class="ot">&lt;-</span> fibrosis<span class="sc">$</span>Status</span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> df <span class="sc">%&gt;%</span></span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>Status, <span class="at">names_to =</span> <span class="st">&quot;Lineage&quot;</span>, <span class="at">values_to =</span> <span class="st">&quot;ws&quot;</span>)</span>
<span id="cb29-5"><a href="#cb29-5" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(df, <span class="fu">aes</span>(<span class="at">x =</span> ws)) <span class="sc">+</span></span>
<span id="cb29-6"><a href="#cb29-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_density</span>(<span class="at">alpha =</span> .<span class="dv">8</span>, <span class="fu">aes</span>(<span class="at">fill =</span> Status), <span class="at">col =</span> <span class="st">&quot;transparent&quot;</span>) <span class="sc">+</span></span>
<span id="cb29-7"><a href="#cb29-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_density</span>(<span class="fu">aes</span>(<span class="at">col =</span> Status), <span class="at">fill =</span> <span class="st">&quot;transparent&quot;</span>, <span class="at">size =</span> <span class="fl">1.5</span>) <span class="sc">+</span></span>
<span id="cb29-8"><a href="#cb29-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;Weights&quot;</span>, <span class="at">fill =</span> <span class="st">&quot;Status&quot;</span>) <span class="sc">+</span></span>
<span id="cb29-9"><a href="#cb29-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>Lineage, <span class="at">scales =</span> <span class="st">&quot;free_x&quot;</span>) <span class="sc">+</span></span>
<span id="cb29-10"><a href="#cb29-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">fill =</span> <span class="st">&quot;none&quot;</span>, <span class="at">col =</span> <span class="st">&quot;none&quot;</span>) <span class="sc">+</span></span>
<span id="cb29-11"><a href="#cb29-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_brewer</span>(<span class="at">palette =</span> <span class="st">&quot;Accent&quot;</span>) <span class="sc">+</span></span>
<span id="cb29-12"><a href="#cb29-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_brewer</span>(<span class="at">palette =</span> <span class="st">&quot;Accent&quot;</span>)</span></code></pre></div>
<p><img src="Fibrosis_files/figure-html/unnamed-chunk-13-1.png" width="700"  style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="differential-expression" class="section level1">
<h1>Differential expression</h1>
<p>We use <em>tradeSeq</em> <span class="citation">(Van den Berge et al. 2020)</span>.</p>
<div id="select-number-of-knots" class="section level2">
<h2>Select number of knots</h2>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tradeSeq)</span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">3</span>)</span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a>BPPARAM <span class="ot">&lt;-</span> BiocParallel<span class="sc">::</span><span class="fu">bpparam</span>()</span>
<span id="cb30-4"><a href="#cb30-4" aria-hidden="true" tabindex="-1"></a>BPPARAM<span class="sc">$</span>workers <span class="ot">&lt;-</span> <span class="dv">3</span></span>
<span id="cb30-5"><a href="#cb30-5" aria-hidden="true" tabindex="-1"></a>icMat <span class="ot">&lt;-</span> <span class="fu">evaluateK</span>(<span class="at">counts =</span> fibrosis, <span class="at">sds =</span> <span class="fu">SlingshotDataSet</span>(fibrosis), </span>
<span id="cb30-6"><a href="#cb30-6" aria-hidden="true" tabindex="-1"></a>                   <span class="at">conditions =</span> <span class="fu">factor</span>(fibrosis<span class="sc">$</span>Status),</span>
<span id="cb30-7"><a href="#cb30-7" aria-hidden="true" tabindex="-1"></a>                   <span class="at">nGenes =</span> <span class="dv">300</span>, <span class="at">parallel =</span> <span class="cn">TRUE</span>, <span class="at">BPPARAM =</span> BPPARAM, <span class="at">k =</span> <span class="dv">3</span><span class="sc">:</span><span class="dv">7</span>)</span></code></pre></div>
<div id="fit-gam" class="section level3">
<h3>Fit GAM</h3>
<p>Next, we fit the NB-GAMs using 6 knots, based on the pseudotime and cell-level weights estimated by Slingshot. We use the <code>conditions</code> argument to fit separate smoothers for each condition.</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tradeSeq)</span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">3</span>)</span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a>fibrosis <span class="ot">&lt;-</span> <span class="fu">fitGAM</span>(<span class="at">counts =</span> fibrosis, <span class="at">conditions =</span> <span class="fu">factor</span>(fibrosis<span class="sc">$</span>Status), <span class="at">nknots =</span> <span class="dv">6</span>)</span></code></pre></div>
</div>
</div>
<div id="differential-expression-between-conditions" class="section level2">
<h2>Differential expression between conditions</h2>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a>condRes <span class="ot">&lt;-</span> <span class="fu">conditionTest</span>(fibrosis, <span class="at">l2fc =</span> <span class="fu">log2</span>(<span class="dv">2</span>), <span class="at">lineages =</span> <span class="cn">TRUE</span>)</span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a>condRes<span class="sc">$</span>padj <span class="ot">&lt;-</span> <span class="fu">p.adjust</span>(condRes<span class="sc">$</span>pvalue_lineage1, <span class="st">&quot;fdr&quot;</span>)</span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a><span class="fu">mean</span>(condRes<span class="sc">$</span>padj <span class="sc">&lt;=</span> <span class="fl">0.05</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<pre><code>## [1] 0.0002972946</code></pre>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(condRes<span class="sc">$</span>padj <span class="sc">&lt;=</span> <span class="fl">0.05</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<pre><code>## [1] 3</code></pre>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a>conditionGenes <span class="ot">&lt;-</span> <span class="fu">rownames</span>(condRes)[condRes<span class="sc">$</span>padj <span class="sc">&lt;=</span> <span class="fl">0.05</span>]</span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a>conditionGenes <span class="ot">&lt;-</span> conditionGenes[<span class="sc">!</span><span class="fu">is.na</span>(conditionGenes)]</span></code></pre></div>
</div>
<div id="differential-expression-between-lineages" class="section level2">
<h2>Differential expression between lineages</h2>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a>patRes <span class="ot">&lt;-</span> <span class="fu">patternTest</span>(fibrosis, <span class="at">l2fc =</span> <span class="fu">log2</span>(<span class="dv">2</span>))</span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a>patRes<span class="sc">$</span>padj <span class="ot">&lt;-</span> <span class="fu">p.adjust</span>(patRes<span class="sc">$</span>pvalue, <span class="st">&quot;fdr&quot;</span>)</span>
<span id="cb37-3"><a href="#cb37-3" aria-hidden="true" tabindex="-1"></a><span class="fu">mean</span>(patRes<span class="sc">$</span>padj <span class="sc">&lt;=</span> <span class="fl">0.05</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<pre><code>## [1] 0.1574202</code></pre>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(patRes<span class="sc">$</span>padj <span class="sc">&lt;=</span> <span class="fl">0.05</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<pre><code>## [1] 1589</code></pre>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a>patternGenes <span class="ot">&lt;-</span> <span class="fu">rownames</span>(patRes)[patRes<span class="sc">$</span>padj <span class="sc">&lt;=</span> <span class="fl">0.05</span>]</span>
<span id="cb41-2"><a href="#cb41-2" aria-hidden="true" tabindex="-1"></a>patternGenes <span class="ot">&lt;-</span> patternGenes[<span class="sc">!</span><span class="fu">is.na</span>(patternGenes)]</span></code></pre></div>
<div id="visualize-select-genes-from-the-og-paper" class="section level3">
<h3>Visualize select genes from the og paper</h3>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(RColorBrewer)</span>
<span id="cb42-2"><a href="#cb42-2" aria-hidden="true" tabindex="-1"></a>scales <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;#FCAE91&quot;</span>, <span class="st">&quot;#DE2D26&quot;</span>, <span class="st">&quot;#BDD7E7&quot;</span>, <span class="st">&quot;#3182BD&quot;</span>)</span>
<span id="cb42-3"><a href="#cb42-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-4"><a href="#cb42-4" aria-hidden="true" tabindex="-1"></a><span class="co"># plot genes</span></span>
<span id="cb42-5"><a href="#cb42-5" aria-hidden="true" tabindex="-1"></a>oo <span class="ot">&lt;-</span> <span class="fu">order</span>(patRes<span class="sc">$</span>waldStat, <span class="at">decreasing =</span> <span class="cn">TRUE</span>)</span>
<span id="cb42-6"><a href="#cb42-6" aria-hidden="true" tabindex="-1"></a>genes <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;SCGB3A2&quot;</span>, <span class="st">&quot;SFTPC&quot;</span>, <span class="st">&quot;ABCA3&quot;</span>, <span class="st">&quot;AGER&quot;</span>)</span>
<span id="cb42-7"><a href="#cb42-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-8"><a href="#cb42-8" aria-hidden="true" tabindex="-1"></a><span class="co"># most significant gene</span></span>
<span id="cb42-9"><a href="#cb42-9" aria-hidden="true" tabindex="-1"></a>ps <span class="ot">&lt;-</span> <span class="fu">lapply</span>(genes, <span class="cf">function</span>(gene){</span>
<span id="cb42-10"><a href="#cb42-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plotSmoothers</span>(fibrosis, <span class="fu">assays</span>(fibrosis)<span class="sc">$</span>counts, <span class="at">gene =</span> gene, <span class="at">curvesCol =</span> scales,</span>
<span id="cb42-11"><a href="#cb42-11" aria-hidden="true" tabindex="-1"></a>                <span class="at">border =</span> <span class="cn">FALSE</span>, <span class="at">sample =</span> .<span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb42-12"><a href="#cb42-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_x_reverse</span>(<span class="at">breaks =</span> <span class="dv">0</span><span class="sc">:</span><span class="dv">4</span><span class="sc">/</span>.<span class="dv">4</span>, <span class="at">labels =</span> <span class="dv">4</span><span class="sc">:</span><span class="dv">0</span><span class="sc">/</span>.<span class="dv">4</span>) <span class="sc">+</span></span>
<span id="cb42-13"><a href="#cb42-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">title =</span> gene, <span class="at">col =</span> <span class="st">&quot;Lineage and Condition&quot;</span>) <span class="sc">+</span></span>
<span id="cb42-14"><a href="#cb42-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_color_manual</span>(<span class="at">values =</span> scales) <span class="sc">+</span></span>
<span id="cb42-15"><a href="#cb42-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> .<span class="dv">5</span>),</span>
<span id="cb42-16"><a href="#cb42-16" aria-hidden="true" tabindex="-1"></a>          <span class="at">rect =</span> <span class="fu">element_blank</span>())</span>
<span id="cb42-17"><a href="#cb42-17" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb42-18"><a href="#cb42-18" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>
<pre><code>## Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
## which will replace the existing scale.
## Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
## which will replace the existing scale.
## Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
## which will replace the existing scale.
## Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
## which will replace the existing scale.</code></pre>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a>legend_gene <span class="ot">&lt;-</span> <span class="fu">get_legend</span>(</span>
<span id="cb44-2"><a href="#cb44-2" aria-hidden="true" tabindex="-1"></a>  ps[[<span class="dv">1</span>]] <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;bottom&quot;</span>) <span class="sc">+</span></span>
<span id="cb44-3"><a href="#cb44-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">guides</span>(<span class="at">col =</span> <span class="fu">guide_legend</span>(<span class="at">title.position =</span> <span class="st">&quot;top&quot;</span>))</span>
<span id="cb44-4"><a href="#cb44-4" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb44-5"><a href="#cb44-5" aria-hidden="true" tabindex="-1"></a>ps <span class="ot">&lt;-</span> <span class="fu">lapply</span>(ps, <span class="cf">function</span>(p) {p <span class="sc">+</span> <span class="fu">guides</span>(<span class="at">col =</span> <span class="st">&quot;none&quot;</span>)})</span>
<span id="cb44-6"><a href="#cb44-6" aria-hidden="true" tabindex="-1"></a>p6 <span class="ot">&lt;-</span> <span class="fu">plot_grid</span>(</span>
<span id="cb44-7"><a href="#cb44-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot_grid</span>(<span class="at">plotlist =</span> ps, <span class="at">ncol =</span> <span class="dv">2</span>, <span class="at">scale =</span> .<span class="dv">95</span>,</span>
<span id="cb44-8"><a href="#cb44-8" aria-hidden="true" tabindex="-1"></a>            <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">&#39;a)&#39;</span>, <span class="st">&#39;b)&#39;</span>, <span class="st">&#39;c)&#39;</span>, <span class="st">&#39;d)&#39;</span>)),</span>
<span id="cb44-9"><a href="#cb44-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot_grid</span>(<span class="cn">NULL</span>, legend_gene, <span class="cn">NULL</span>, <span class="at">rel_widths =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>), <span class="at">ncol =</span> <span class="dv">3</span>), </span>
<span id="cb44-10"><a href="#cb44-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">nrow =</span> <span class="dv">2</span>, <span class="at">rel_heights =</span> <span class="fu">c</span>(<span class="dv">4</span>, .<span class="dv">5</span>)</span>
<span id="cb44-11"><a href="#cb44-11" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>
</div>
<div id="heatmaps-of-genes-de-between-lineages" class="section level3">
<h3>Heatmaps of genes DE between lineages</h3>
<p>Below we show heatmaps of the genes DE between lineages The DE genes in the heatmaps are ordered according to a hierarchical clustering on the first lineage.</p>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a><span class="do">### based on mean smoother</span></span>
<span id="cb45-2"><a href="#cb45-2" aria-hidden="true" tabindex="-1"></a>yhatSmooth <span class="ot">&lt;-</span> <span class="fu">predictSmooth</span>(fibrosis, <span class="at">gene =</span> patternGenes, <span class="at">nPoints =</span> <span class="dv">50</span>, <span class="at">tidy =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb45-3"><a href="#cb45-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">yhat =</span> <span class="fu">log1p</span>(yhat)) <span class="sc">%&gt;%</span></span>
<span id="cb45-4"><a href="#cb45-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(gene) <span class="sc">%&gt;%</span></span>
<span id="cb45-5"><a href="#cb45-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">yhat =</span> scales<span class="sc">::</span><span class="fu">rescale</span>(yhat)) <span class="sc">%&gt;%</span></span>
<span id="cb45-6"><a href="#cb45-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(condition <span class="sc">==</span> <span class="st">&quot;ILD&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb45-7"><a href="#cb45-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>condition) <span class="sc">%&gt;%</span></span>
<span id="cb45-8"><a href="#cb45-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>()</span>
<span id="cb45-9"><a href="#cb45-9" aria-hidden="true" tabindex="-1"></a>heatSmooth_Lineage1 <span class="ot">&lt;-</span> <span class="fu">pheatmap</span>(</span>
<span id="cb45-10"><a href="#cb45-10" aria-hidden="true" tabindex="-1"></a>  yhatSmooth <span class="sc">%&gt;%</span></span>
<span id="cb45-11"><a href="#cb45-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(lineage <span class="sc">==</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb45-12"><a href="#cb45-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select</span>(<span class="sc">-</span>lineage) <span class="sc">%&gt;%</span></span>
<span id="cb45-13"><a href="#cb45-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">arrange</span>(<span class="sc">-</span>time) <span class="sc">%&gt;%</span></span>
<span id="cb45-14"><a href="#cb45-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> time, <span class="at">values_from =</span> yhat) <span class="sc">%&gt;%</span></span>
<span id="cb45-15"><a href="#cb45-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select</span>(<span class="sc">-</span>gene),</span>
<span id="cb45-16"><a href="#cb45-16" aria-hidden="true" tabindex="-1"></a>  <span class="at">cluster_cols =</span> <span class="cn">FALSE</span>, <span class="at">show_rownames =</span> <span class="cn">FALSE</span>, <span class="at">show_colnames =</span> <span class="cn">FALSE</span>,</span>
<span id="cb45-17"><a href="#cb45-17" aria-hidden="true" tabindex="-1"></a>  <span class="at">main =</span> <span class="st">&quot;Lineage 1&quot;</span>, <span class="at">legend =</span> <span class="cn">FALSE</span>, <span class="at">silent =</span> <span class="cn">TRUE</span></span>
<span id="cb45-18"><a href="#cb45-18" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb45-19"><a href="#cb45-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-20"><a href="#cb45-20" aria-hidden="true" tabindex="-1"></a>heatSmooth_Lineage2 <span class="ot">&lt;-</span> <span class="fu">pheatmap</span>(</span>
<span id="cb45-21"><a href="#cb45-21" aria-hidden="true" tabindex="-1"></a>  (yhatSmooth <span class="sc">%&gt;%</span></span>
<span id="cb45-22"><a href="#cb45-22" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(lineage <span class="sc">==</span> <span class="dv">2</span>) <span class="sc">%&gt;%</span></span>
<span id="cb45-23"><a href="#cb45-23" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select</span>(<span class="sc">-</span>lineage) <span class="sc">%&gt;%</span></span>
<span id="cb45-24"><a href="#cb45-24" aria-hidden="true" tabindex="-1"></a>    <span class="fu">arrange</span>(<span class="sc">-</span>time) <span class="sc">%&gt;%</span></span>
<span id="cb45-25"><a href="#cb45-25" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> time, <span class="at">values_from =</span> yhat) <span class="sc">%&gt;%</span></span>
<span id="cb45-26"><a href="#cb45-26" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select</span>(<span class="sc">-</span>gene))[heatSmooth_Lineage1<span class="sc">$</span>tree_row<span class="sc">$</span>order, ],</span>
<span id="cb45-27"><a href="#cb45-27" aria-hidden="true" tabindex="-1"></a>  <span class="at">cluster_cols =</span> <span class="cn">FALSE</span>, <span class="at">cluster_rows =</span> <span class="cn">FALSE</span>,</span>
<span id="cb45-28"><a href="#cb45-28" aria-hidden="true" tabindex="-1"></a>  <span class="at">show_rownames =</span> <span class="cn">FALSE</span>, <span class="at">show_colnames =</span> <span class="cn">FALSE</span>, <span class="at">main =</span> <span class="st">&quot;Lineage 2&quot;</span>,</span>
<span id="cb45-29"><a href="#cb45-29" aria-hidden="true" tabindex="-1"></a>  <span class="at">legend =</span> <span class="cn">FALSE</span>, <span class="at">silent =</span> <span class="cn">TRUE</span> </span>
<span id="cb45-30"><a href="#cb45-30" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb45-31"><a href="#cb45-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-32"><a href="#cb45-32" aria-hidden="true" tabindex="-1"></a>p7 <span class="ot">&lt;-</span> <span class="fu">plot_grid</span>(heatSmooth_Lineage1[[<span class="dv">4</span>]], heatSmooth_Lineage2[[<span class="dv">4</span>]], <span class="at">ncol =</span> <span class="dv">2</span>)</span>
<span id="cb45-33"><a href="#cb45-33" aria-hidden="true" tabindex="-1"></a>p7</span></code></pre></div>
<p><img src="Fibrosis_files/figure-html/unnamed-chunk-18-1.png" width="700"  style="display: block; margin: auto;" /></p>
</div>
</div>
</div>
<div id="session-info" class="section level1">
<h1>Session info</h1>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code></pre></div>
<pre><code>## R version 4.1.0 (2021-05-18)
## Platform: x86_64-apple-darwin17.0 (64-bit)
## Running under: macOS Big Sur 10.16
## 
## Matrix products: default
## BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.dylib
## LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] parallel  stats4    stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
##  [1] RColorBrewer_1.1-2          caret_6.0-88               
##  [3] lattice_0.20-44             scales_1.1.1               
##  [5] msigdbr_7.4.1               fgsea_1.18.0               
##  [7] pheatmap_1.0.12             tradeSeq_1.7.04            
##  [9] readr_2.0.0                 condiments_1.1.04          
## [11] cowplot_1.1.1               slingshot_2.1.1            
## [13] TrajectoryUtils_1.0.0       princurve_2.1.6            
## [15] scater_1.20.1               ggplot2_3.3.5              
## [17] scuttle_1.2.0               SingleCellExperiment_1.14.1
## [19] SummarizedExperiment_1.22.0 Biobase_2.52.0             
## [21] GenomicRanges_1.44.0        GenomeInfoDb_1.28.2        
## [23] IRanges_2.26.0              S4Vectors_0.30.0           
## [25] BiocGenerics_0.38.0         MatrixGenerics_1.4.3       
## [27] matrixStats_0.60.1          SeuratObject_4.0.2         
## [29] Seurat_4.0.3                dplyr_1.0.7                
## [31] tidyr_1.1.3                 here_1.0.1                 
## [33] knitr_1.33                 
## 
## loaded via a namespace (and not attached):
##   [1] utf8_1.2.2                reticulate_1.20          
##   [3] tidyselect_1.1.1          htmlwidgets_1.5.3        
##   [5] grid_4.1.0                BiocParallel_1.26.2      
##   [7] Rtsne_0.15                pROC_1.17.0.1            
##   [9] munsell_0.5.0             ScaledMatrix_1.0.0       
##  [11] codetools_0.2-18          ragg_1.1.3               
##  [13] ica_1.0-2                 future_1.21.0            
##  [15] miniUI_0.1.1.1            withr_2.4.2              
##  [17] colorspace_2.0-2          highr_0.9                
##  [19] rstudioapi_0.13           ROCR_1.0-11              
##  [21] tensor_1.5                listenv_0.8.0            
##  [23] labeling_0.4.2            GenomeInfoDbData_1.2.6   
##  [25] polyclip_1.10-0           farver_2.1.0             
##  [27] rprojroot_2.0.2           parallelly_1.27.0        
##  [29] vctrs_0.3.8               generics_0.1.0           
##  [31] ipred_0.9-11              xfun_0.24                
##  [33] Ecume_0.9.1               randomForest_4.6-14      
##  [35] R6_2.5.1                  ggbeeswarm_0.6.0         
##  [37] rsvd_1.0.5                locfit_1.5-9.4           
##  [39] bitops_1.0-7              spatstat.utils_2.2-0     
##  [41] cachem_1.0.5              DelayedArray_0.18.0      
##  [43] assertthat_0.2.1          promises_1.2.0.1         
##  [45] nnet_7.3-16               beeswarm_0.4.0           
##  [47] gtable_0.3.0              beachmat_2.8.0           
##  [49] globals_0.14.0            goftest_1.2-2            
##  [51] timeDate_3043.102         rlang_0.4.11             
##  [53] systemfonts_1.0.2         splines_4.1.0            
##  [55] lazyeval_0.2.2            ModelMetrics_1.2.2.2     
##  [57] spatstat.geom_2.2-2       yaml_2.2.1               
##  [59] reshape2_1.4.4            abind_1.4-5              
##  [61] httpuv_1.6.1              tools_4.1.0              
##  [63] lava_1.6.9                ellipsis_0.3.2           
##  [65] spatstat.core_2.3-0       jquerylib_0.1.4          
##  [67] proxy_0.4-26              ggridges_0.5.3           
##  [69] Rcpp_1.0.7                plyr_1.8.6               
##  [71] sparseMatrixStats_1.4.0   zlibbioc_1.38.0          
##  [73] purrr_0.3.4               RCurl_1.98-1.4           
##  [75] rpart_4.1-15              deldir_0.2-10            
##  [77] pbapply_1.4-3             viridis_0.6.1            
##  [79] zoo_1.8-9                 ggrepel_0.9.1            
##  [81] cluster_2.1.2             fs_1.5.0                 
##  [83] magrittr_2.0.1            data.table_1.14.0        
##  [85] scattermore_0.7           lmtest_0.9-38            
##  [87] RANN_2.6.1                fitdistrplus_1.1-5       
##  [89] hms_1.1.0                 patchwork_1.1.1          
##  [91] mime_0.11                 evaluate_0.14            
##  [93] xtable_1.8-4              gridExtra_2.3            
##  [95] transport_0.12-2          compiler_4.1.0           
##  [97] tibble_3.1.4              KernSmooth_2.23-20       
##  [99] crayon_1.4.1              htmltools_0.5.1.1        
## [101] tzdb_0.1.2                mgcv_1.8-36              
## [103] later_1.2.0               lubridate_1.7.10         
## [105] DBI_1.1.1                 MASS_7.3-54              
## [107] babelgene_21.4            Matrix_1.3-4             
## [109] cli_3.0.1                 gower_0.2.2              
## [111] igraph_1.2.6              pkgconfig_2.0.3          
## [113] pkgdown_1.6.1             plotly_4.9.4.1           
## [115] spatstat.sparse_2.0-0     recipes_0.1.16           
## [117] foreach_1.5.1             vipor_0.4.5              
## [119] bslib_0.2.5.1             XVector_0.32.0           
## [121] prodlim_2019.11.13        stringr_1.4.0            
## [123] digest_0.6.27             sctransform_0.3.2        
## [125] RcppAnnoy_0.0.18          spatstat.data_2.1-0      
## [127] fastmatch_1.1-0           rmarkdown_2.9            
## [129] leiden_0.3.8              edgeR_3.34.0             
## [131] uwot_0.1.10               DelayedMatrixStats_1.14.0
## [133] kernlab_0.9-29            shiny_1.6.0              
## [135] lifecycle_1.0.0           nlme_3.1-152             
## [137] jsonlite_1.7.2            BiocNeighbors_1.10.0     
## [139] limma_3.48.3              desc_1.3.0               
## [141] viridisLite_0.4.0         fansi_0.5.0              
## [143] spatstat.linnet_2.3-0     pillar_1.6.2             
## [145] fastmap_1.1.0             httr_1.4.2               
## [147] survival_3.2-11           glue_1.4.2               
## [149] spatstat_2.2-0            png_0.1-7                
## [151] iterators_1.0.13          class_7.3-19             
## [153] stringi_1.7.3             sass_0.4.0               
## [155] textshaping_0.3.5         BiocSingular_1.8.1       
## [157] memoise_2.0.0             e1071_1.7-7              
## [159] irlba_2.3.3               future.apply_1.7.0</code></pre>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="unnumbered">References</h1>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-Street2018a" class="csl-entry">
Street, Kelly, Davide Risso, Russell B. Fletcher, Diya Das, John Ngai, Nir Yosef, Elizabeth Purdom, and Sandrine Dudoit. 2018. <span>“<span class="nocase">Slingshot: cell lineage and pseudotime inference for single-cell transcriptomics</span>.”</span> <em>BMC Genomics</em> 19 (1): 477. <a href="https://doi.org/10.1186/s12864-018-4772-0">https://doi.org/10.1186/s12864-018-4772-0</a>.
</div>
<div id="ref-VandenBerge2020" class="csl-entry">
Van den Berge, Koen, Hector Roux de Bézieux, Kelly Street, Wouter Saelens, Robrecht Cannoodt, Yvan Saeys, Sandrine Dudoit, and Lieven Clement. 2020. <span>“<span class="nocase">Trajectory-based differential expression analysis for single-cell sequencing data</span>.”</span> <em>Nature Communications</em> 11 (1): 1201. <a href="https://doi.org/10.1038/s41467-020-14766-3">https://doi.org/10.1038/s41467-020-14766-3</a>.
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc">
      <h2 data-toc-skip>Contents</h2>
    </nav>
      </div>

</div>



      <footer>
      <div class="copyright">
  <p>Developed by Hector Roux de Bezieux, Koen Van den Berge, Kelly Street.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>
