<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Analysis of the TCDD datataset • condimentsPaper</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Analysis of the TCDD datataset">
<meta property="og:description" content="condimentsPaper">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">condimentsPaper</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/Examples.html">Figure 1: examples</a>
    </li>
    <li>
      <a href="../articles/KRAS.html">Analysis of the KRAS datataset</a>
    </li>
    <li>
      <a href="../articles/Simulations.html">Running and analyzing the simulations</a>
    </li>
    <li>
      <a href="../articles/TCDD.html">Analysis of the TCDD datataset</a>
    </li>
    <li>
      <a href="../articles/TGFB.html">Analysis of the TGFB datataset</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/HectorRDB/condimentsPaper">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Analysis of the TCDD datataset</h1>
                        <h4 class="author">Hector Roux de Bézieux</h4>
            
      
      
      <div class="hidden name"><code>TCDD.Rmd</code></div>

    </div>

    
    
<p>The dataset we will be working with concerns a single-cell RNA-sequencing dataset consisting of two different experiments, which correspond to treatment (TCDD) and control. Nault et al. <span class="citation">(Nault et al. 2021)</span> studied the effect of that drug on the liver, along the central-portal axis.</p>
<div id="load-data" class="section level1">
<h1 class="hasAnchor">
<a href="#load-data" class="anchor"></a>Load data</h1>
<div class="sourceCode" id="cb1"><html><body><pre class="r"><span class="no">cds</span> <span class="kw">&lt;-</span> <span class="kw pkg">condimentsPaper</span><span class="kw ns">::</span><span class="fu"><a href="../reference/import_TCDD.html">import_TCDD</a></span>()
<span class="co"># Get zonal genes ----</span>
<span class="no">url</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span>(<span class="st">"https://static-content.springer.com/esm/art%3A10.1038%2Fnature"</span>,
              <span class="st">"21065/MediaObjects/41586_2017_BFnature21065_MOESM62_ESM.xlsx"</span>)
<span class="no">df</span> <span class="kw">&lt;-</span> <span class="kw pkg">openxlsx</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/openxlsx/man/read.xlsx.html">read.xlsx</a></span>(<span class="no">url</span>, <span class="kw">startRow</span> <span class="kw">=</span> <span class="fl">3</span>, <span class="kw">na.strings</span> <span class="kw">=</span> <span class="st">"NaN"</span>)
<span class="no">df</span> <span class="kw">&lt;-</span> <span class="no">df</span>[, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">1</span>, <span class="fl">21</span>)] <span class="kw">%&gt;%</span>
  <span class="kw pkg">dplyr</span><span class="kw ns">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/rename.html">rename</a></span>(<span class="st">"qvalues"</span> <span class="kw">=</span> <span class="no">`q-values`</span>) <span class="kw">%&gt;%</span>
  <span class="kw pkg">dplyr</span><span class="kw ns">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span>(!<span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span>(<span class="no">qvalues</span>), <span class="no">qvalues</span> <span class="kw">&lt;</span> <span class="fl">.005</span>) <span class="kw">%&gt;%</span>
  <span class="kw pkg">dplyr</span><span class="kw ns">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span>(<span class="no">Gene.Symbol</span>)
<span class="no">zone.mat</span> <span class="kw">&lt;-</span> <span class="no">df</span>$<span class="no">Gene.Symbol</span> <span class="kw">%&gt;%</span> <span class="kw pkg">stringr</span><span class="kw ns">::</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/str_split.html">str_split</a></span>(<span class="st">";"</span>) <span class="kw">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/unlist.html">unlist</a></span>()
<span class="co"># Filter to only keep hepatocytes</span>
<span class="no">hepatocytes</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span>(<span class="no">cds</span>, <span class="kw">subset</span> <span class="kw">=</span> <span class="no">celltype.ontology</span> <span class="kw">==</span> <span class="st">"CL:0000182:Hepatocyte"</span>)
<span class="no">common.genes</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sets.html">intersect</a></span>(<span class="no">zone.mat</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="no">hepatocytes</span>))
<span class="fu">DefaultAssay</span>(<span class="no">hepatocytes</span>) <span class="kw">&lt;-</span> <span class="st">"integrated"</span>
<span class="no">hepatocytes</span> <span class="kw">&lt;-</span> <span class="fu">ScaleData</span>(<span class="no">hepatocytes</span>, <span class="kw">verbose</span> <span class="kw">=</span> <span class="fl">FALSE</span>)
<span class="no">hepatocytes</span> <span class="kw">&lt;-</span> <span class="fu">RunPCA</span>(<span class="no">hepatocytes</span>, <span class="kw">features</span> <span class="kw">=</span> <span class="no">common.genes</span>, <span class="kw">verbose</span> <span class="kw">=</span> <span class="fl">FALSE</span>)
<span class="no">hepatocytes</span> <span class="kw">&lt;-</span> <span class="fu">RunUMAP</span>(<span class="no">hepatocytes</span>, <span class="kw">dims</span> <span class="kw">=</span> <span class="fl">1</span>:<span class="fl">30</span>, <span class="kw">verbose</span> <span class="kw">=</span> <span class="fl">FALSE</span>)
<span class="no">tcdd</span> <span class="kw">&lt;-</span> <span class="fu">as.SingleCellExperiment</span>(<span class="no">hepatocytes</span>, <span class="kw">assay</span> <span class="kw">=</span> <span class="st">"RNA"</span>)
<span class="no">tcdd</span>$<span class="no">celltype</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/droplevels.html">droplevels</a></span>(<span class="no">tcdd</span>$<span class="no">celltype</span>) <span class="kw">%&gt;%</span>
  <span class="kw pkg">stringr</span><span class="kw ns">::</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/str_remove.html">str_remove</a></span>(<span class="st">"Hepatocytes - "</span>) <span class="kw">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/r/base/unlist.html">unlist</a></span>()
<span class="fu">rowData</span>(<span class="no">tcdd</span>)$<span class="no">is_zonal</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="no">tcdd</span>) <span class="kw">%in%</span> <span class="no">zone.mat</span></pre></body></html></div>
<div class="sourceCode" id="cb2"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="st">"tcdd"</span>, <span class="kw">package</span> <span class="kw">=</span> <span class="st">"condimentsPaper"</span>)</pre></body></html></div>
</div>
<div id="eda" class="section level1">
<h1 class="hasAnchor">
<a href="#eda" class="anchor"></a>EDA</h1>
<p>The data is imported using a function from the package. We then compute reduced dimension coordinates with UMAP <span class="citation">(Becht et al. 2019, @McInnes2018)</span>.</p>
<div class="sourceCode" id="cb3"><html><body><pre class="r"><span class="no">df</span> <span class="kw">&lt;-</span> <span class="fu">bind_cols</span>(
  <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span>(<span class="fu">reducedDims</span>(<span class="no">tcdd</span>)$<span class="no">UMAP</span>),
  <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span>(<span class="fu">colData</span>(<span class="no">tcdd</span>)))
<span class="no">p1</span> <span class="kw">&lt;-</span> <span class="fu">ggplot</span>(<span class="no">df</span>, <span class="fu">aes</span>(<span class="kw">x</span> <span class="kw">=</span> <span class="no">UMAP_1</span>, <span class="kw">y</span> <span class="kw">=</span> <span class="no">UMAP_2</span>, <span class="kw">col</span> <span class="kw">=</span> <span class="no">treatment</span>)) +
  <span class="fu">geom_point</span>(<span class="kw">size</span> <span class="kw">=</span> <span class="fl">.7</span>) +
  <span class="fu">scale_color_brewer</span>(<span class="kw">palette</span> <span class="kw">=</span> <span class="st">"Accent"</span>) +
  <span class="fu">labs</span>(<span class="kw">col</span> <span class="kw">=</span> <span class="st">"Treatment"</span>)
<span class="no">p2</span> <span class="kw">&lt;-</span> <span class="fu">ggplot</span>(<span class="no">df</span>, <span class="fu">aes</span>(<span class="kw">x</span> <span class="kw">=</span> <span class="no">UMAP_1</span>, <span class="kw">y</span> <span class="kw">=</span> <span class="no">UMAP_2</span>, <span class="kw">fill</span> <span class="kw">=</span> <span class="no">celltype</span>)) +
  <span class="fu">geom_point</span>(<span class="kw">size</span> <span class="kw">=</span> <span class="fl">1</span>, <span class="kw">alpha</span> <span class="kw">=</span> <span class="fl">.65</span>, <span class="kw">col</span> <span class="kw">=</span> <span class="st">"grey70"</span>, <span class="kw">shape</span> <span class="kw">=</span> <span class="fl">21</span>) +
  <span class="fu">scale_fill_manual</span>(<span class="kw">values</span> <span class="kw">=</span> <span class="no">pal</span>) +
  <span class="fu">labs</span>(<span class="kw">fill</span> <span class="kw">=</span> <span class="st">"Cell Type"</span>)
<span class="no">p1</span></pre></body></html></div>
<p><img src="TCDD_files/figure-html/unnamed-chunk-4-1.png" width="700" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb4"><html><body><pre class="r"><span class="no">p2</span></pre></body></html></div>
<p><img src="TCDD_files/figure-html/unnamed-chunk-4-2.png" width="700" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb5"><html><body><pre class="r"><span class="no">tcdd</span> <span class="kw">&lt;-</span> <span class="fu">imbalance_score</span>(<span class="no">tcdd</span>, <span class="kw">dimred</span> <span class="kw">=</span> <span class="st">"UMAP"</span>, <span class="kw">conditions</span> <span class="kw">=</span> <span class="st">"treatment"</span>, <span class="kw">smooth</span> <span class="kw">=</span> <span class="fl">5</span>)
<span class="no">df</span>$<span class="no">scores</span> <span class="kw">&lt;-</span> <span class="no">tcdd</span>$<span class="no">scores</span>$<span class="no">scaled_scores</span>
<span class="no">p3</span> <span class="kw">&lt;-</span> <span class="fu">ggplot</span>(<span class="no">df</span>, <span class="fu">aes</span>(<span class="kw">x</span> <span class="kw">=</span> <span class="no">UMAP_1</span>, <span class="kw">y</span> <span class="kw">=</span> <span class="no">UMAP_2</span>, <span class="kw">col</span> <span class="kw">=</span> <span class="no">scores</span>)) +
  <span class="fu">geom_point</span>(<span class="kw">size</span> <span class="kw">=</span> <span class="fl">.7</span>) +
  <span class="fu">scale_color_viridis_c</span>(<span class="kw">option</span> <span class="kw">=</span> <span class="st">"C"</span>) +
  <span class="fu">labs</span>(<span class="kw">col</span> <span class="kw">=</span> <span class="st">"Score"</span>)
<span class="no">p3</span></pre></body></html></div>
<p><img src="TCDD_files/figure-html/unnamed-chunk-5-1.png" width="700" style="display: block; margin: auto;"></p>
</div>
<div id="differential-topology" class="section level1">
<h1 class="hasAnchor">
<a href="#differential-topology" class="anchor"></a>Differential Topology</h1>
<p>To estimate the trajectory, we use <em>slingshot</em> <span class="citation">(Street et al. 2018)</span>.</p>
<div class="sourceCode" id="cb6"><html><body><pre class="r"><span class="no">sds</span> <span class="kw">&lt;-</span> <span class="fu">slingshot</span>(<span class="fu">reducedDim</span>(<span class="no">tcdd</span>, <span class="st">"UMAP"</span>),
                 <span class="kw">clusterLabels</span> <span class="kw">=</span> <span class="no">tcdd</span>$<span class="no">celltype</span>,
                 <span class="kw">start.clus</span> <span class="kw">=</span> <span class="st">"Central"</span>,
                 <span class="kw">end.clus</span>  <span class="kw">=</span> <span class="st">"Portal"</span>)</pre></body></html></div>
<pre><code>## Using full covariance matrix</code></pre>
<div class="sourceCode" id="cb8"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span>(<span class="fl">9764</span>)
<span class="fu">topologyTest</span>(<span class="no">sds</span>, <span class="no">tcdd</span>$<span class="no">treatment</span>, <span class="kw">rep</span> <span class="kw">=</span> <span class="fl">100</span>, <span class="kw">methods</span> <span class="kw">=</span> <span class="st">"KS_mean"</span>, <span class="kw">threshs</span> <span class="kw">=</span> <span class="fl">.01</span>)</pre></body></html></div>
<pre><code>## Generating permuted trajectories</code></pre>
<pre><code>## Running KS-mean test</code></pre>
<pre><code>##    method thresh  statistic    p.value
## 1 KS_mean   0.01 0.01823837 0.07302285</code></pre>
<div class="sourceCode" id="cb12"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="no">df</span>) <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(<span class="no">tcdd</span>)
<span class="no">pst</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span>(<span class="kw">cells</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="no">sds</span>@<span class="kw">reducedDim</span>),
                  <span class="kw">pst</span> <span class="kw">=</span> <span class="fu">slingPseudotime</span>(<span class="no">sds</span>)[, <span class="fl">1</span>])
<span class="no">df</span>$<span class="no">cells</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="no">df</span>)
<span class="no">df</span> <span class="kw">&lt;-</span> <span class="kw pkg">dplyr</span><span class="kw ns">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html">full_join</a></span>(<span class="no">df</span>, <span class="no">pst</span>)</pre></body></html></div>
<pre><code>## Joining, by = "cells"</code></pre>
<div class="sourceCode" id="cb14"><html><body><pre class="r"><span class="no">p4</span> <span class="kw">&lt;-</span> <span class="fu">ggplot</span>(<span class="no">df</span>, <span class="fu">aes</span>(<span class="kw">x</span> <span class="kw">=</span> <span class="no">UMAP_1</span>, <span class="kw">y</span> <span class="kw">=</span> <span class="no">UMAP_2</span>, <span class="kw">col</span> <span class="kw">=</span> <span class="no">pst</span>)) +
  <span class="fu">geom_point</span>(<span class="kw">size</span> <span class="kw">=</span> <span class="fl">.7</span>) +
  <span class="fu">scale_color_viridis_c</span>() +
  <span class="fu">labs</span>(<span class="kw">col</span> <span class="kw">=</span> <span class="st">"Pseudotime"</span>) +
  <span class="fu">geom_path</span>(<span class="kw">data</span> <span class="kw">=</span> <span class="no">sds</span>@<span class="kw">curves</span>$<span class="no">curve1</span>$<span class="no">s</span>[<span class="no">sds</span>@<span class="kw">curves</span>$<span class="no">curve1</span>$<span class="no">ord</span>, ] <span class="kw">%&gt;%</span>
              <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span>() <span class="kw">%&gt;%</span>
              <span class="kw pkg">dplyr</span><span class="kw ns">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span>(<span class="kw">X</span> <span class="kw">=</span> <span class="no">UMAP_1</span>, <span class="kw">Y</span> <span class="kw">=</span> <span class="no">UMAP_2</span>),
            <span class="kw">col</span> <span class="kw">=</span> <span class="st">"black"</span>, <span class="kw">size</span> <span class="kw">=</span> <span class="fl">1.5</span>)
<span class="no">p4</span></pre></body></html></div>
<p><img src="TCDD_files/figure-html/unnamed-chunk-8-1.png" width="700" style="display: block; margin: auto;"></p>
</div>
<div id="differential-progression" class="section level1">
<h1 class="hasAnchor">
<a href="#differential-progression" class="anchor"></a>Differential Progression</h1>
<div class="sourceCode" id="cb15"><html><body><pre class="r"><span class="no">p5</span> <span class="kw">&lt;-</span> <span class="fu">ggplot</span>(<span class="no">df</span>, <span class="fu">aes</span>(<span class="kw">x</span> <span class="kw">=</span> <span class="no">pst</span>)) +
  <span class="fu">geom_density</span>(<span class="kw">alpha</span> <span class="kw">=</span> <span class="fl">.8</span>, <span class="fu">aes</span>(<span class="kw">fill</span> <span class="kw">=</span> <span class="no">treatment</span>), <span class="kw">col</span> <span class="kw">=</span> <span class="st">"transparent"</span>) +
  <span class="fu">geom_density</span>(<span class="fu">aes</span>(<span class="kw">col</span> <span class="kw">=</span> <span class="no">treatment</span>), <span class="kw">fill</span> <span class="kw">=</span> <span class="st">"transparent"</span>, <span class="kw">size</span> <span class="kw">=</span> <span class="fl">1.5</span>) +
  <span class="fu">guides</span>(<span class="kw">col</span> <span class="kw">=</span> <span class="fl">FALSE</span>,
         <span class="kw">fill</span> <span class="kw">=</span> <span class="fu">guide_legend</span>(
             <span class="kw">override.aes</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span>(<span class="kw">size</span> <span class="kw">=</span> <span class="fl">1.5</span>,
                                 <span class="kw">col</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"#7FC97F"</span>, <span class="st">"#BEAED4"</span>),
                                <span class="kw">title.position</span> <span class="kw">=</span> <span class="st">"top"</span>)
         )) +
  <span class="fu">scale_fill_brewer</span>(<span class="kw">palette</span> <span class="kw">=</span> <span class="st">"Accent"</span>) +
  <span class="fu">scale_color_brewer</span>(<span class="kw">palette</span> <span class="kw">=</span> <span class="st">"Accent"</span>) +
  <span class="fu">labs</span>(<span class="kw">x</span> <span class="kw">=</span> <span class="st">"Pseudotime"</span>, <span class="kw">fill</span> <span class="kw">=</span> <span class="st">"Treatment"</span>)
<span class="no">p5</span></pre></body></html></div>
<p><img src="TCDD_files/figure-html/unnamed-chunk-9-1.png" width="700" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb16"><html><body><pre class="r"><span class="fu">progressionTest</span>(<span class="no">sds</span>, <span class="kw">conditions</span> <span class="kw">=</span> <span class="no">tcdd</span>$<span class="no">treatment</span>)</pre></body></html></div>
<pre><code>## Registered S3 method overwritten by 'cli':
##   method     from    
##   print.boxx spatstat</code></pre>
<pre><code>## # A tibble: 1 x 3
##   lineage statistic  p.value
##   &lt;chr&gt;       &lt;dbl&gt;    &lt;dbl&gt;
## 1 1           0.267 2.20e-16</code></pre>
</div>
<div id="differential-expression" class="section level1">
<h1 class="hasAnchor">
<a href="#differential-expression" class="anchor"></a>Differential Expression</h1>
<p>We use <em>tradeSeq</em> <span class="citation">(Van den Berge et al. 2020)</span>.</p>
<div class="sourceCode" id="cb19"><html><body><pre class="r"><span class="no">filter</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html">apply</a></span>(<span class="fu">counts</span>(<span class="no">tcdd</span>), <span class="fl">1</span>, <span class="kw">function</span>(<span class="no">g</span>) {
    <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span>(<span class="no">g</span> <span class="kw">&gt;=</span> <span class="fl">3</span>) <span class="kw">&gt;=</span> <span class="fl">10</span>
})
<span class="no">tcdd</span> <span class="kw">&lt;-</span> <span class="no">tcdd</span>[<span class="no">filter</span>, ]</pre></body></html></div>
<div class="sourceCode" id="cb20"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span>(<span class="fl">3</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">BiocParallel</span>)
<span class="no">BPPARAM</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocParallel/man/register.html">bpparam</a></span>()
<span class="no">BPPARAM</span>$<span class="no">workers</span> <span class="kw">&lt;-</span> <span class="fl">5</span>
<span class="no">BPPARAM</span>$<span class="no">progressbar</span> <span class="kw">&lt;-</span> <span class="fl">TRUE</span>
<span class="no">icMat</span> <span class="kw">&lt;-</span> <span class="fu">evaluateK</span>(<span class="kw">counts</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span>(<span class="fu">assays</span>(<span class="no">tcdd</span>)$<span class="no">counts</span>),
                   <span class="kw">pseudotime</span> <span class="kw">=</span> <span class="fu">slingPseudotime</span>(<span class="no">sds</span>),
                   <span class="kw">cellWeights</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span>(<span class="fl">1</span>, <span class="fu"><a href="https://rdrr.io/r/base/nrow.html">ncol</a></span>(<span class="no">tcdd</span>)),
                   <span class="kw">conditions</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span>(<span class="no">tcdd</span>$<span class="no">treatment</span>),
                   <span class="kw">nGenes</span> <span class="kw">=</span> <span class="fl">300</span>,
                   <span class="kw">k</span> <span class="kw">=</span> <span class="fl">3</span>:<span class="fl">7</span>, <span class="kw">parallel</span> <span class="kw">=</span> <span class="fl">TRUE</span>,
                   <span class="kw">BPPARAM</span> <span class="kw">=</span> <span class="no">BPPARAM</span>)
<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span>(<span class="fl">3</span>)
<span class="no">tcdd</span>@<span class="kw">int_metadata</span>$<span class="no">slingshot</span> <span class="kw">&lt;-</span> <span class="no">sds</span>
<span class="no">tcdd</span> <span class="kw">&lt;-</span> <span class="fu">fitGAM</span>(<span class="kw">counts</span> <span class="kw">=</span> <span class="no">tcdd</span>,
              <span class="kw">conditions</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span>(<span class="no">tcdd</span>$<span class="no">treatment</span>),
              <span class="kw">nknots</span> <span class="kw">=</span> <span class="fl">7</span>,
              <span class="kw">verbose</span> <span class="kw">=</span> <span class="fl">TRUE</span>,
              <span class="kw">parallel</span> <span class="kw">=</span> <span class="fl">TRUE</span>)</pre></body></html></div>
<div id="assessing-differential-expression-accross-conditions" class="section level2">
<h2 class="hasAnchor">
<a href="#assessing-differential-expression-accross-conditions" class="anchor"></a>Assessing Differential Expression Accross conditions</h2>
<div class="sourceCode" id="cb21"><html><body><pre class="r"><span class="no">condRes</span> <span class="kw">&lt;-</span> <span class="fu">conditionTest</span>(<span class="no">tcdd</span>, <span class="kw">l2fc</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html">log2</a></span>(<span class="fl">2</span>))
<span class="no">condRes</span>$<span class="no">padj</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/p.adjust.html">p.adjust</a></span>(<span class="no">condRes</span>$<span class="no">pvalue</span>, <span class="st">"fdr"</span>)
<span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span>(<span class="no">condRes</span>$<span class="no">padj</span> <span class="kw">&lt;=</span> <span class="fl">0.05</span>, <span class="kw">na.rm</span> <span class="kw">=</span> <span class="fl">TRUE</span>)</pre></body></html></div>
<pre><code>## [1] 0.2634268</code></pre>
<div class="sourceCode" id="cb23"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span>(<span class="no">condRes</span>$<span class="no">padj</span> <span class="kw">&lt;=</span> <span class="fl">0.05</span>, <span class="kw">na.rm</span> <span class="kw">=</span> <span class="fl">TRUE</span>)</pre></body></html></div>
<pre><code>## [1] 2114</code></pre>
<div class="sourceCode" id="cb25"><html><body><pre class="r"><span class="no">conditionGenes</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="no">condRes</span>)[<span class="no">condRes</span>$<span class="no">padj</span> <span class="kw">&lt;=</span> <span class="fl">0.05</span>]
<span class="no">conditionGenes</span> <span class="kw">&lt;-</span> <span class="no">conditionGenes</span>[!<span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span>(<span class="no">conditionGenes</span>)]
<span class="no">scales</span> <span class="kw">&lt;-</span> <span class="fu">brewer.pal</span>(<span class="fl">3</span>, <span class="st">"Accent"</span>)[<span class="fl">1</span>:<span class="fl">2</span>]
<span class="no">oo</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/order.html">order</a></span>(<span class="no">condRes</span>$<span class="no">waldStat</span>, <span class="kw">decreasing</span> <span class="kw">=</span> <span class="fl">TRUE</span>)

<span class="co"># most significant gene</span>
<span class="no">p1</span> <span class="kw">&lt;-</span> <span class="fu">plotSmoothers</span>(<span class="no">tcdd</span>, <span class="fu">assays</span>(<span class="no">tcdd</span>)$<span class="no">counts</span>,
                    <span class="kw">gene</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="fu">assays</span>(<span class="no">tcdd</span>)$<span class="no">counts</span>)[<span class="no">oo</span>[<span class="fl">1</span>]],
                    <span class="kw">alpha</span> <span class="kw">=</span> <span class="fl">1</span>, <span class="kw">border</span> <span class="kw">=</span> <span class="fl">TRUE</span>, <span class="kw">curvesCols</span> <span class="kw">=</span> <span class="no">scales</span>) +
  <span class="fu">scale_color_manual</span>(<span class="kw">values</span> <span class="kw">=</span> <span class="no">scales</span>) +
  <span class="fu">ggtitle</span>(<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="fu">assays</span>(<span class="no">tcdd</span>)$<span class="no">counts</span>)[<span class="no">oo</span>[<span class="fl">1</span>]])</pre></body></html></div>
<pre><code>## Scale for 'colour' is already present. Adding another scale for 'colour',
## which will replace the existing scale.</code></pre>
<div class="sourceCode" id="cb27"><html><body><pre class="r"><span class="co"># second most significant gene</span>
<span class="no">p2</span> <span class="kw">&lt;-</span> <span class="fu">plotSmoothers</span>(<span class="no">tcdd</span>, <span class="fu">assays</span>(<span class="no">tcdd</span>)$<span class="no">counts</span>,
                    <span class="kw">gene</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="fu">assays</span>(<span class="no">tcdd</span>)$<span class="no">counts</span>)[<span class="no">oo</span>[<span class="fl">2</span>]],
                    <span class="kw">alpha</span> <span class="kw">=</span> <span class="fl">1</span>, <span class="kw">border</span> <span class="kw">=</span> <span class="fl">TRUE</span>, <span class="kw">curvesCols</span> <span class="kw">=</span> <span class="no">scales</span>) +
  <span class="fu">scale_color_manual</span>(<span class="kw">values</span> <span class="kw">=</span> <span class="no">scales</span>) +
  <span class="fu">ggtitle</span>(<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="fu">assays</span>(<span class="no">tcdd</span>)$<span class="no">counts</span>)[<span class="no">oo</span>[<span class="fl">2</span>]])</pre></body></html></div>
<pre><code>## Scale for 'colour' is already present. Adding another scale for 'colour',
## which will replace the existing scale.</code></pre>
<div class="sourceCode" id="cb29"><html><body><pre class="r"><span class="co"># least significant gene</span>
<span class="no">p3</span> <span class="kw">&lt;-</span> <span class="fu">plotSmoothers</span>(<span class="no">tcdd</span>, <span class="fu">assays</span>(<span class="no">tcdd</span>)$<span class="no">counts</span>,
                    <span class="kw">gene</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="fu">assays</span>(<span class="no">tcdd</span>)$<span class="no">counts</span>)[<span class="no">oo</span>[<span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span>(<span class="no">tcdd</span>)]],
                    <span class="kw">alpha</span> <span class="kw">=</span> <span class="fl">1</span>, <span class="kw">border</span> <span class="kw">=</span> <span class="fl">TRUE</span>, <span class="kw">curvesCols</span> <span class="kw">=</span> <span class="no">scales</span>) +
  <span class="fu">scale_color_manual</span>(<span class="kw">values</span> <span class="kw">=</span> <span class="no">scales</span>) +
  <span class="fu">ggtitle</span>(<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="fu">assays</span>(<span class="no">tcdd</span>)$<span class="no">counts</span>)[<span class="no">oo</span>[<span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span>(<span class="no">tcdd</span>)]])</pre></body></html></div>
<pre><code>## Scale for 'colour' is already present. Adding another scale for 'colour',
## which will replace the existing scale.</code></pre>
<div class="sourceCode" id="cb31"><html><body><pre class="r"><span class="no">p1</span></pre></body></html></div>
<p><img src="TCDD_files/figure-html/unnamed-chunk-13-1.png" width="700" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb32"><html><body><pre class="r"><span class="no">p2</span></pre></body></html></div>
<p><img src="TCDD_files/figure-html/unnamed-chunk-13-2.png" width="700" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb33"><html><body><pre class="r"><span class="no">p3</span></pre></body></html></div>
<p><img src="TCDD_files/figure-html/unnamed-chunk-13-3.png" width="700" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb34"><html><body><pre class="r"><span class="co">### based on mean smoother</span>
<span class="no">yhatSmooth</span> <span class="kw">&lt;-</span> <span class="fu">predictSmooth</span>(<span class="no">tcdd</span>, <span class="kw">gene</span> <span class="kw">=</span> <span class="no">conditionGenes</span>, <span class="kw">nPoints</span> <span class="kw">=</span> <span class="fl">50</span>, <span class="kw">tidy</span> <span class="kw">=</span> <span class="fl">FALSE</span>) <span class="kw">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/r/base/Log.html">log1p</a></span>()
<span class="no">yhatSmoothScaled</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/t.html">t</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/apply.html">apply</a></span>(<span class="no">yhatSmooth</span>,<span class="fl">1</span>, <span class="kw pkg">scales</span><span class="kw ns">::</span><span class="no"><a href="https://scales.r-lib.org//reference/rescale.html">rescale</a></span>))
<span class="no">heatSmooth_TCDD</span> <span class="kw">&lt;-</span> <span class="fu">pheatmap</span>(<span class="no">yhatSmoothScaled</span>[, <span class="fl">51</span>:<span class="fl">100</span>],
  <span class="kw">cluster_cols</span> <span class="kw">=</span> <span class="fl">FALSE</span>,
  <span class="kw">show_rownames</span> <span class="kw">=</span> <span class="fl">FALSE</span>, <span class="kw">show_colnames</span> <span class="kw">=</span> <span class="fl">FALSE</span>, <span class="kw">main</span> <span class="kw">=</span> <span class="st">"TCDD"</span>, <span class="kw">legend</span> <span class="kw">=</span> <span class="fl">FALSE</span>,
  <span class="kw">silent</span> <span class="kw">=</span> <span class="fl">TRUE</span>
)

<span class="no">matchingHeatmap_control</span> <span class="kw">&lt;-</span> <span class="fu">pheatmap</span>(<span class="no">yhatSmoothScaled</span>[<span class="no">heatSmooth_TCDD</span>$<span class="no">tree_row</span>$<span class="no">order</span>, <span class="fl">1</span>:<span class="fl">50</span>],
  <span class="kw">cluster_cols</span> <span class="kw">=</span> <span class="fl">FALSE</span>, <span class="kw">cluster_rows</span> <span class="kw">=</span> <span class="fl">FALSE</span>,
  <span class="kw">show_rownames</span> <span class="kw">=</span> <span class="fl">FALSE</span>, <span class="kw">show_colnames</span> <span class="kw">=</span> <span class="fl">FALSE</span>, <span class="kw">main</span> <span class="kw">=</span> <span class="st">"Control"</span>,
  <span class="kw">legend</span> <span class="kw">=</span> <span class="fl">FALSE</span>, <span class="kw">silent</span> <span class="kw">=</span> <span class="fl">TRUE</span>
)

<span class="no">p4</span> <span class="kw">&lt;-</span> <span class="fu">plot_grid</span>(<span class="no">heatSmooth_TCDD</span><span class="kw">[[</span><span class="fl">4</span>]], <span class="no">matchingHeatmap_control</span><span class="kw">[[</span><span class="fl">4</span>]], <span class="kw">ncol</span> <span class="kw">=</span> <span class="fl">2</span>)
<span class="no">p4</span></pre></body></html></div>
<p><img src="TCDD_files/figure-html/unnamed-chunk-14-1.png" width="700" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb35"><html><body><pre class="r"><span class="no">kept</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="no">tcdd</span>)[<span class="fu">rowData</span>(<span class="no">tcdd</span>)$<span class="no">is_zonal</span>]
<span class="no">mat</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">matrix</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span>(<span class="no">kept</span> <span class="kw">%in%</span> <span class="no">conditionGenes</span>), <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="no">conditionGenes</span>),
                <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span>(<span class="no">kept</span> <span class="kw">%in%</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="no">tcdd</span>)[!<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="no">tcdd</span>) <span class="kw">%in%</span> <span class="no">conditionGenes</span>]),
                <span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span>(<span class="no">tcdd</span>) - <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="no">conditionGenes</span>)),
              <span class="kw">byrow</span> <span class="kw">=</span> <span class="fl">FALSE</span>, <span class="kw">ncol</span> <span class="kw">=</span> <span class="fl">2</span>)
<span class="fu"><a href="https://rdrr.io/r/stats/fisher.test.html">fisher.test</a></span>(<span class="no">mat</span>)</pre></body></html></div>
<pre><code>## 
##  Fisher's Exact Test for Count Data
## 
## data:  mat
## p-value &lt; 2.2e-16
## alternative hypothesis: true odds ratio is not equal to 1
## 95 percent confidence interval:
##  1.752901 2.396818
## sample estimates:
## odds ratio 
##   2.050607</code></pre>
</div>
</div>
<div id="session-info" class="section level1">
<h1 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session info</h1>
<div class="sourceCode" id="cb37"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span>()</pre></body></html></div>
<pre><code>## R version 4.0.3 (2020-10-10)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 20.04.1 LTS
## 
## Matrix products: default
## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3
## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/liblapack.so.3
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] parallel  stats4    stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
##  [1] scales_1.1.1                pheatmap_1.0.12            
##  [3] RColorBrewer_1.1-2          tradeSeq_1.5.07            
##  [5] cowplot_1.0.0               ggplot2_3.3.1              
##  [7] condiments_0.1.0            slingshot_1.6.1            
##  [9] princurve_2.1.5             SingleCellExperiment_1.10.1
## [11] SummarizedExperiment_1.18.2 DelayedArray_0.14.1        
## [13] matrixStats_0.57.0          Biobase_2.48.0             
## [15] GenomicRanges_1.40.0        GenomeInfoDb_1.24.2        
## [17] IRanges_2.22.2              S4Vectors_0.26.1           
## [19] BiocGenerics_0.34.0         Seurat_3.2.3               
## [21] stringr_1.4.0               dplyr_1.0.0                
## [23] knitr_1.28                 
## 
## loaded via a namespace (and not attached):
##   [1] utf8_1.1.4             reticulate_1.16        tidyselect_1.1.0      
##   [4] htmlwidgets_1.5.1      grid_4.0.3             combinat_0.0-8        
##   [7] docopt_0.7.1           BiocParallel_1.22.0    Rtsne_0.15            
##  [10] pROC_1.16.2            munsell_0.5.0          codetools_0.2-18      
##  [13] ica_1.0-2              future_1.17.0          miniUI_0.1.1.1        
##  [16] withr_2.3.0            colorspace_2.0-0       fastICA_1.2-2         
##  [19] rstudioapi_0.13        ROCR_1.0-11            tensor_1.5            
##  [22] listenv_0.8.0          labeling_0.4.2         slam_0.1-48           
##  [25] GenomeInfoDbData_1.2.3 polyclip_1.10-0        farver_2.0.3          
##  [28] rprojroot_2.0.2        vctrs_0.3.1            generics_0.1.0        
##  [31] ipred_0.9-9            xfun_0.14              Ecume_0.9.0           
##  [34] R6_2.5.0               rsvd_1.0.3             VGAM_1.1-4            
##  [37] locfit_1.5-9.4         bitops_1.0-6           spatstat.utils_1.17-0 
##  [40] assertthat_0.2.1       promises_1.1.0         nnet_7.3-14           
##  [43] gtable_0.3.0           globals_0.12.5         goftest_1.2-2         
##  [46] timeDate_3043.102      rlang_0.4.10           splines_4.0.3         
##  [49] lazyeval_0.2.2         ModelMetrics_1.2.2.2   yaml_2.2.1            
##  [52] reshape2_1.4.4         abind_1.4-5            httpuv_1.5.3.1        
##  [55] caret_6.0-86           tools_4.0.3            lava_1.6.7            
##  [58] ellipsis_0.3.1         ggridges_0.5.2         Rcpp_1.0.5            
##  [61] plyr_1.8.6             zlibbioc_1.34.0        purrr_0.3.4           
##  [64] RCurl_1.98-1.2         densityClust_0.3       rpart_4.1-15          
##  [67] deldir_0.1-25          pbapply_1.4-3          viridis_0.5.1         
##  [70] zoo_1.8-8              ggrepel_0.8.2          cluster_2.1.0         
##  [73] fs_1.4.1               magrittr_2.0.1         data.table_1.12.8     
##  [76] scattermore_0.7        lmtest_0.9-37          RANN_2.6.1            
##  [79] fitdistrplus_1.1-1     patchwork_1.0.0        mime_0.9              
##  [82] evaluate_0.14          xtable_1.8-4           sparsesvd_0.2         
##  [85] gridExtra_2.3          HSMMSingleCell_1.8.0   compiler_4.0.3        
##  [88] transport_0.12-2       tibble_3.0.1           KernSmooth_2.23-18    
##  [91] crayon_1.3.4           htmltools_0.4.0        mgcv_1.8-33           
##  [94] later_1.1.0.1          tidyr_1.1.0            lubridate_1.7.8       
##  [97] MASS_7.3-51.6          rappdirs_0.3.1         Matrix_1.2-18         
## [100] cli_2.0.2              gower_0.2.1            igraph_1.2.5          
## [103] pkgconfig_2.0.3        pkgdown_1.5.1          plotly_4.9.2.1        
## [106] recipes_0.1.12         foreach_1.5.0          XVector_0.28.0        
## [109] prodlim_2019.11.13     digest_0.6.27          sctransform_0.3.2     
## [112] RcppAnnoy_0.0.18       DDRTree_0.1.5          spatstat.data_1.4-3   
## [115] rmarkdown_2.2          leiden_0.3.3           uwot_0.1.10           
## [118] edgeR_3.30.3           kernlab_0.9-29         shiny_1.4.0.2         
## [121] lifecycle_0.2.0        monocle_2.16.0         nlme_3.1-151          
## [124] jsonlite_1.7.2         fansi_0.4.1            desc_1.2.0            
## [127] viridisLite_0.3.0      limma_3.44.3           pillar_1.4.4          
## [130] lattice_0.20-41        fastmap_1.0.1          httr_1.4.1            
## [133] survival_3.1-12        glue_1.4.2             qlcMatrix_0.9.7       
## [136] FNN_1.1.3              spatstat_1.64-1        png_0.1-7             
## [139] iterators_1.0.12       class_7.3-17           stringi_1.5.3         
## [142] memoise_1.1.0          irlba_2.3.3            e1071_1.7-3           
## [145] future.apply_1.5.0     ape_5.4-1</code></pre>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references">
<div id="ref-Becht2019">
<p>Becht, Etienne, Leland McInnes, John Healy, Charles-Antoine Dutertre, Immanuel W H Kwok, Lai Guan Ng, Florent Ginhoux, and Evan W Newell. 2019. “Dimensionality reduction for visualizing single-cell data using UMAP.” <em>Nature Biotechnology</em> 37 (1): 38–44. <a href="https://doi.org/10.1038/nbt.4314">https://doi.org/10.1038/nbt.4314</a>.</p>
</div>
<div id="ref-McInnes2018">
<p>McInnes, Leland, John Healy, and James Melville. 2018. “UMAP: Uniform Manifold Approximation and Projection for Dimension Reduction.” <em>Arxiv</em>, February. <a href="http://arxiv.org/abs/1802.03426">http://arxiv.org/abs/1802.03426</a>.</p>
</div>
<div id="ref-Nault2021">
<p>Nault, Rance, Kelly A. Fader, Sudin Bhattacharya, and Tim R. Zacharewski. 2021. “Single-Nuclei RNA Sequencing Assessment of the Hepatic Effects of 2,3,7,8-Tetrachlorodibenzo-p-dioxin.” <em>CMGH</em> 11 (1): 147–59. <a href="https://doi.org/10.1016/j.jcmgh.2020.07.012">https://doi.org/10.1016/j.jcmgh.2020.07.012</a>.</p>
</div>
<div id="ref-Street2018a">
<p>Street, Kelly, Davide Risso, Russell B. Fletcher, Diya Das, John Ngai, Nir Yosef, Elizabeth Purdom, and Sandrine Dudoit. 2018. “Slingshot: cell lineage and pseudotime inference for single-cell transcriptomics.” <em>BMC Genomics</em> 19 (1): 477. <a href="https://doi.org/10.1186/s12864-018-4772-0">https://doi.org/10.1186/s12864-018-4772-0</a>.</p>
</div>
<div id="ref-VandenBerge2020">
<p>Van den Berge, Koen, Hector Roux de Bézieux, Kelly Street, Wouter Saelens, Robrecht Cannoodt, Yvan Saeys, Sandrine Dudoit, and Lieven Clement. 2020. “Trajectory-based differential expression analysis for single-cell sequencing data.” <em>Nature Communications</em> 11 (1): 1201. <a href="https://doi.org/10.1038/s41467-020-14766-3">https://doi.org/10.1038/s41467-020-14766-3</a>.</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Hector Roux de Bezieux, Koen Van den Berge, Kelly Street.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
