<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Analysis of the TGFB datataset • condimentsPaper</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Analysis of the TGFB datataset">
<meta property="og:description" content="condimentsPaper">
<meta property="og:image" content="http://hectorrdb.github.io//condimentsPaper/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">condimentsPaper</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/Examples.html">Figure 1: examples</a>
    </li>
    <li>
      <a href="../articles/KRAS.html">Analysis of the KRAS datataset</a>
    </li>
    <li>
      <a href="../articles/Simulations.html">Running and analyzing the simulations</a>
    </li>
    <li>
      <a href="../articles/TCDD.html">Analysis of the TCDD datataset</a>
    </li>
    <li>
      <a href="../articles/TGFB.html">Analysis of the TGFB datataset</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/HectorRDB/condimentsPaper">
    <span class="fas fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="TGFB_files/header-attrs-2.7/header-attrs.js"></script><script src="TGFB_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Analysis of the TGFB datataset</h1>
                        <h4 class="author">Hector Roux de Bézieux, Koen Van den Berge and Kelly Street</h4>
            
      
      
      <div class="hidden name"><code>TGFB.Rmd</code></div>

    </div>

    
    
<div id="overview" class="section level1">
<h1 class="hasAnchor">
<a href="#overview" class="anchor"></a>Overview</h1>
<p>We reproduce a lot of code from the <a href="https://kstreet13.github.io/bioc2020trajectories/articles/workshopTrajectories.html">Bioc 2020 trajectory workshop</a>.</p>
</div>
<div id="dataset" class="section level1">
<h1 class="hasAnchor">
<a href="#dataset" class="anchor"></a>Dataset</h1>
<p>The dataset we will be working with concerns a single-cell RNA-sequencing dataset consisting of two different experiments, which correspond to two treatments. McFaline-Figueroa et al. <span class="citation">(McFaline-Figueroa et al. 2019)</span> studied the epithelial-to-mesenchymal transition (EMT), where cells spatially migrate from the epithelium to the mesenchyme during development.</p>
</div>
<div id="load-data" class="section level1">
<h1 class="hasAnchor">
<a href="#load-data" class="anchor"></a>Load data</h1>
<p>The data is imported using a function from the package. We then normalized using <em>Seurat</em><span class="citation">(Stuart et al. 2019)</span> and compute reduced dimension coordinates with UMAP <span class="citation">(Becht et al. 2019, @McInnes2018)</span>.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">tgfb</span> <span class="op">&lt;-</span> <span class="fu">condimentsPaper</span><span class="fu">::</span><span class="fu"><a href="../reference/import_TGFB.html">import_TGFB</a></span><span class="op">(</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat">Seurat</a></span><span class="op">)</span>
<span class="co">########################</span>
<span class="co">### Split by condition and convert to Seurat</span>
<span class="co">########################</span>
<span class="fu">assays</span><span class="op">(</span><span class="va">tgfb</span><span class="op">)</span><span class="op">$</span><span class="va">logcounts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html">log1p</a></span><span class="op">(</span><span class="fu">assays</span><span class="op">(</span><span class="va">tgfb</span><span class="op">)</span><span class="op">$</span><span class="va">counts</span><span class="op">)</span>
<span class="va">tgfbMock</span> <span class="op">&lt;-</span> <span class="va">tgfb</span><span class="op">[</span> ,<span class="fu">colData</span><span class="op">(</span><span class="va">tgfb</span><span class="op">)</span><span class="op">$</span><span class="va">pheno</span><span class="op">$</span><span class="va">treatment_id</span><span class="op">==</span><span class="st">'Mock'</span><span class="op">]</span>
<span class="va">tgfbTGFB</span> <span class="op">&lt;-</span> <span class="va">tgfb</span><span class="op">[</span> ,<span class="fu">colData</span><span class="op">(</span><span class="va">tgfb</span><span class="op">)</span><span class="op">$</span><span class="va">pheno</span><span class="op">$</span><span class="va">treatment_id</span><span class="op">==</span><span class="st">'TGFB'</span><span class="op">]</span>
<span class="va">soMock</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/as.Seurat.html">as.Seurat</a></span><span class="op">(</span><span class="va">tgfbMock</span><span class="op">)</span>
<span class="va">soTGFB</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/as.Seurat.html">as.Seurat</a></span><span class="op">(</span><span class="va">tgfbTGFB</span><span class="op">)</span>

<span class="co">########################</span>
<span class="co">### Normalize</span>
<span class="co">########################</span>
<span class="va">soMock</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/SCTransform.html">SCTransform</a></span><span class="op">(</span><span class="va">soMock</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="va">soTGFB</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/SCTransform.html">SCTransform</a></span><span class="op">(</span><span class="va">soTGFB</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>

<span class="co">########################</span>
<span class="co">### Integrate</span>
<span class="co">########################</span>
<span class="va">dtlist</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>Mock <span class="op">=</span> <span class="va">soMock</span>, TGFB <span class="op">=</span> <span class="va">soTGFB</span><span class="op">)</span>
<span class="va">intfts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/SelectIntegrationFeatures.html">SelectIntegrationFeatures</a></span><span class="op">(</span>object.list <span class="op">=</span> <span class="va">dtlist</span>, nfeatures <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="va">tgfb</span><span class="op">)</span><span class="op">)</span> <span class="co"># maxes out at 4080 (why?)</span>
<span class="va">dtlist</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/PrepSCTIntegration.html">PrepSCTIntegration</a></span><span class="op">(</span>object.list <span class="op">=</span> <span class="va">dtlist</span>,
                             anchor.features <span class="op">=</span> <span class="va">intfts</span><span class="op">)</span>
<span class="va">anchors</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/FindIntegrationAnchors.html">FindIntegrationAnchors</a></span><span class="op">(</span>object.list <span class="op">=</span> <span class="va">dtlist</span>, normalization.method <span class="op">=</span> <span class="st">"SCT"</span>,
                                  anchor.features <span class="op">=</span> <span class="va">intfts</span><span class="op">)</span>
<span class="va">integrated</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/IntegrateData.html">IntegrateData</a></span><span class="op">(</span>anchorset <span class="op">=</span> <span class="va">anchors</span>, normalization.method <span class="op">=</span> <span class="st">"SCT"</span><span class="op">)</span>
<span class="va">integrated</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/RunPCA.html">RunPCA</a></span><span class="op">(</span><span class="va">integrated</span><span class="op">)</span>
<span class="va">integrated</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/RunUMAP.html">RunUMAP</a></span><span class="op">(</span><span class="va">integrated</span>, dims <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">50</span><span class="op">)</span>

<span class="co">## convert back to singleCellExperiment</span>
<span class="va">tgfb</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/as.SingleCellExperiment.html">as.SingleCellExperiment</a></span><span class="op">(</span><span class="va">integrated</span>, assay <span class="op">=</span> <span class="st">"RNA"</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"tgfb"</span>, package <span class="op">=</span> <span class="st">"condimentsPaper"</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">df</span> <span class="op">&lt;-</span> <span class="fu">bind_cols</span><span class="op">(</span>
  <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="fu">reducedDims</span><span class="op">(</span><span class="va">tgfb</span><span class="op">)</span><span class="op">$</span><span class="va">UMAP</span><span class="op">)</span>,
  <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="fu">colData</span><span class="op">(</span><span class="va">tgfb</span><span class="op">)</span><span class="op">)</span>
  <span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">sample_frac</span><span class="op">(</span><span class="fl">1</span><span class="op">)</span>
<span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">df</span>, <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">UMAP_1</span>, y <span class="op">=</span> <span class="va">UMAP_2</span>, col <span class="op">=</span> <span class="va">pheno.treatment_id</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">geom_point</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">.7</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">scale_color_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Accent"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">labs</span><span class="op">(</span>col <span class="op">=</span> <span class="st">"Treatment"</span><span class="op">)</span>
<span class="va">p1</span></code></pre></div>
<p><img src="TGFB_files/figure-html/compute%20scores-1.png" width="672"></p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">df</span>, <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">UMAP_1</span>, y <span class="op">=</span> <span class="va">UMAP_2</span>, col <span class="op">=</span> <span class="va">pheno.spatial_id</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">geom_point</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">.7</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">scale_color_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Dark2"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">labs</span><span class="op">(</span>col <span class="op">=</span> <span class="st">"Spatial ID"</span><span class="op">)</span>
<span class="va">p2</span></code></pre></div>
<p><img src="TGFB_files/figure-html/unnamed-chunk-4-1.png" width="672"></p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">scores</span> <span class="op">&lt;-</span> <span class="fu">condiments</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/condiments/man/imbalance_score.html">imbalance_score</a></span><span class="op">(</span>
  Object <span class="op">=</span> <span class="va">df</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">UMAP_1</span>, <span class="va">UMAP_2</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span><span class="op">(</span><span class="op">)</span>, 
  conditions <span class="op">=</span> <span class="va">df</span><span class="op">$</span><span class="va">pheno.treatment_id</span>,
  k <span class="op">=</span> <span class="fl">20</span>, smooth <span class="op">=</span> <span class="fl">40</span><span class="op">)</span>
<span class="va">df</span><span class="op">$</span><span class="va">scores</span> <span class="op">&lt;-</span> <span class="va">scores</span><span class="op">$</span><span class="va">scaled_scores</span>
<span class="va">p3</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">df</span>, <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">UMAP_1</span>, y <span class="op">=</span> <span class="va">UMAP_2</span>, col <span class="op">=</span> <span class="va">scores</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">geom_point</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">.7</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">scale_color_viridis_c</span><span class="op">(</span>option <span class="op">=</span> <span class="st">"C"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">labs</span><span class="op">(</span>col <span class="op">=</span> <span class="st">"Scores"</span><span class="op">)</span>
<span class="va">p3</span></code></pre></div>
<p><img src="TGFB_files/figure-html/unnamed-chunk-5-1.png" width="672"></p>
</div>
<div id="trajectory-inference-and-differential-topology" class="section level1">
<h1 class="hasAnchor">
<a href="#trajectory-inference-and-differential-topology" class="anchor"></a>Trajectory Inference and Differential Topology</h1>
<p>To estimate the trajectory, we use <em>slingshot</em> <span class="citation">(Street et al. 2018)</span>.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">slingshot</span><span class="op">)</span>
<span class="va">tgfb</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/slingshot/man/slingshot.html">slingshot</a></span><span class="op">(</span><span class="va">tgfb</span>, reducedDim <span class="op">=</span> <span class="st">'UMAP'</span>,
                  clusterLabels <span class="op">=</span> <span class="fu">colData</span><span class="op">(</span><span class="va">tgfb</span><span class="op">)</span><span class="op">$</span><span class="va">pheno</span><span class="op">$</span><span class="va">spatial_id</span>,
                  start.clus <span class="op">=</span> <span class="st">'inner'</span>, approx_points <span class="op">=</span> <span class="fl">100</span><span class="op">)</span></code></pre></div>
<pre><code>## Using full covariance matrix</code></pre>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">821</span><span class="op">)</span>
<span class="fu">topologyTest</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/slingshot/man/SlingshotDataSet.html">SlingshotDataSet</a></span><span class="op">(</span><span class="va">tgfb</span><span class="op">)</span>, <span class="va">tgfb</span><span class="op">$</span><span class="va">pheno</span><span class="op">$</span><span class="va">treatment_id</span>, rep <span class="op">=</span> <span class="fl">100</span>,
             methods <span class="op">=</span> <span class="st">"KS_mean"</span>, threshs <span class="op">=</span> <span class="fl">.01</span><span class="op">)</span></code></pre></div>
<pre><code>## Generating permuted trajectories</code></pre>
<pre><code>## Running KS-mean test</code></pre>
<pre><code>##    method thresh statistic   p.value
## 1 KS_mean   0.01  0.013306 0.3847918</code></pre>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">df</span> <span class="op">&lt;-</span> <span class="fu">bind_cols</span><span class="op">(</span>
  <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="fu">reducedDims</span><span class="op">(</span><span class="va">tgfb</span><span class="op">)</span><span class="op">$</span><span class="va">UMAP</span><span class="op">)</span>,
  <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="fu">colData</span><span class="op">(</span><span class="va">tgfb</span><span class="op">)</span><span class="op">)</span>
  <span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">sample_frac</span><span class="op">(</span><span class="fl">1</span><span class="op">)</span>
<span class="va">curve</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/slingshot/man/slingCurves.html">slingCurves</a></span><span class="op">(</span><span class="va">tgfb</span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>
<span class="va">p4</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">df</span>, <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">UMAP_1</span>, y <span class="op">=</span> <span class="va">UMAP_2</span>, col <span class="op">=</span> <span class="va">slingPseudotime_1</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">geom_point</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">.7</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">scale_color_viridis_c</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">labs</span><span class="op">(</span>col <span class="op">=</span> <span class="st">"Pseudotime"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">geom_path</span><span class="op">(</span>data <span class="op">=</span> <span class="va">curve</span><span class="op">$</span><span class="va">s</span><span class="op">[</span><span class="va">curve</span><span class="op">$</span><span class="va">ord</span>, <span class="op">]</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="op">)</span>,
            col <span class="op">=</span> <span class="st">"black"</span>, size <span class="op">=</span> <span class="fl">1.5</span><span class="op">)</span>
<span class="va">p4</span></code></pre></div>
<p><img src="TGFB_files/figure-html/plotSlingshot-1.png" width="672"></p>
</div>
<div id="differential-progression" class="section level1">
<h1 class="hasAnchor">
<a href="#differential-progression" class="anchor"></a>Differential progression</h1>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p5</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">df</span>, <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">slingPseudotime_1</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">geom_density</span><span class="op">(</span>alpha <span class="op">=</span> <span class="fl">.8</span>, <span class="fu">aes</span><span class="op">(</span>fill <span class="op">=</span> <span class="va">pheno.treatment_id</span><span class="op">)</span>, col <span class="op">=</span> <span class="st">"transparent"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">geom_density</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>col <span class="op">=</span> <span class="va">pheno.treatment_id</span><span class="op">)</span>, fill <span class="op">=</span> <span class="st">"transparent"</span>,
               guide <span class="op">=</span> <span class="cn">FALSE</span>, size <span class="op">=</span> <span class="fl">1.5</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">labs</span><span class="op">(</span>x <span class="op">=</span> <span class="st">"Pseudotime"</span>, fill <span class="op">=</span> <span class="st">"Treatment"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">guides</span><span class="op">(</span>col <span class="op">=</span> <span class="cn">FALSE</span>, fill <span class="op">=</span> <span class="fu">guide_legend</span><span class="op">(</span>
    override.aes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">1.5</span>, col <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"#7FC97F"</span>, <span class="st">"#BEAED4"</span><span class="op">)</span><span class="op">)</span>
  <span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">scale_fill_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Accent"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">scale_color_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Accent"</span><span class="op">)</span></code></pre></div>
<pre><code>## Warning: Ignoring unknown parameters: guide</code></pre>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p5</span></code></pre></div>
<p><img src="TGFB_files/figure-html/plotDensities-1.png" width="672"></p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">progressionTest</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/slingshot/man/SlingshotDataSet.html">SlingshotDataSet</a></span><span class="op">(</span><span class="va">tgfb</span><span class="op">)</span>, conditions <span class="op">=</span> <span class="va">tgfb</span><span class="op">$</span><span class="va">pheno</span><span class="op">$</span><span class="va">treatment_id</span><span class="op">)</span></code></pre></div>
<pre><code>## Registered S3 method overwritten by 'cli':
##   method     from    
##   print.boxx spatstat</code></pre>
<pre><code>## # A tibble: 1 x 3
##   lineage statistic  p.value
##   &lt;chr&gt;       &lt;dbl&gt;    &lt;dbl&gt;
## 1 1           0.176 2.20e-16</code></pre>
</div>
<div id="differential-expression" class="section level1">
<h1 class="hasAnchor">
<a href="#differential-expression" class="anchor"></a>Differential expression</h1>
<p>We use <em>tradeSeq</em> <span class="citation">(Van den Berge et al. 2020)</span>.</p>
<div id="select-number-of-knots" class="section level2">
<h2 class="hasAnchor">
<a href="#select-number-of-knots" class="anchor"></a>Select number of knots</h2>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://statomics.github.io/tradeSeq/index.html">tradeSeq</a></span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">3</span><span class="op">)</span>
<span class="va">icMat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/tradeSeq/man/evaluateK.html">evaluateK</a></span><span class="op">(</span>counts <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span><span class="op">(</span><span class="fu">assays</span><span class="op">(</span><span class="va">tgfb</span><span class="op">)</span><span class="op">$</span><span class="va">counts</span><span class="op">)</span>,
                   pseudotime <span class="op">=</span> <span class="fu">colData</span><span class="op">(</span><span class="va">tgfb</span><span class="op">)</span><span class="op">$</span><span class="va">slingshot</span><span class="op">$</span><span class="va">pseudotime</span>,
                   cellWeights <span class="op">=</span> <span class="fu">colData</span><span class="op">(</span><span class="va">tgfb</span><span class="op">)</span><span class="op">$</span><span class="va">slingshot</span><span class="op">$</span><span class="va">cellWeights.V1</span>,
                   conditions <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="fu">colData</span><span class="op">(</span><span class="va">tgfb</span><span class="op">)</span><span class="op">$</span><span class="va">pheno</span><span class="op">$</span><span class="va">treatment_id</span><span class="op">)</span>,
                   nGenes <span class="op">=</span> <span class="fl">300</span>,
                   k <span class="op">=</span> <span class="fl">3</span><span class="op">:</span><span class="fl">7</span><span class="op">)</span></code></pre></div>
</div>
<div id="fit-gam" class="section level2">
<h2 class="hasAnchor">
<a href="#fit-gam" class="anchor"></a>Fit GAM</h2>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">3</span><span class="op">)</span>
<span class="va">tgfb</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/tradeSeq/man/fitGAM.html">fitGAM</a></span><span class="op">(</span>counts <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span><span class="op">(</span><span class="fu">assays</span><span class="op">(</span><span class="va">tgfb</span><span class="op">)</span><span class="op">$</span><span class="va">counts</span><span class="op">)</span>,
               pseudotime <span class="op">=</span> <span class="fu">colData</span><span class="op">(</span><span class="va">tgfb</span><span class="op">)</span><span class="op">$</span><span class="va">slingshot</span><span class="op">$</span><span class="va">pseudotime</span>,
               cellWeights <span class="op">=</span> <span class="fu">colData</span><span class="op">(</span><span class="va">tgfb</span><span class="op">)</span><span class="op">$</span><span class="va">slingshot</span><span class="op">$</span><span class="va">cellWeights.V1</span>,
               conditions <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="fu">colData</span><span class="op">(</span><span class="va">tgfb</span><span class="op">)</span><span class="op">$</span><span class="va">pheno</span><span class="op">$</span><span class="va">treatment_id</span><span class="op">)</span>,
               nknots <span class="op">=</span> <span class="fl">5</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="fu">rowData</span><span class="op">(</span><span class="va">tgfb</span><span class="op">)</span><span class="op">$</span><span class="va">tradeSeq</span><span class="op">$</span><span class="va">converged</span><span class="op">)</span></code></pre></div>
</div>
<div id="differential-expression-between-conditions" class="section level2">
<h2 class="hasAnchor">
<a href="#differential-expression-between-conditions" class="anchor"></a>Differential expression between conditions</h2>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">condRes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/tradeSeq/man/conditionTest.html">conditionTest</a></span><span class="op">(</span><span class="va">tgfb</span>, l2fc <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html">log2</a></span><span class="op">(</span><span class="fl">2</span><span class="op">)</span><span class="op">)</span>
<span class="va">condRes</span><span class="op">$</span><span class="va">padj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/p.adjust.html">p.adjust</a></span><span class="op">(</span><span class="va">condRes</span><span class="op">$</span><span class="va">pvalue</span>, <span class="st">"fdr"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">condRes</span><span class="op">$</span><span class="va">padj</span> <span class="op">&lt;=</span> <span class="fl">0.05</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 0.1889995</code></pre>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">condRes</span><span class="op">$</span><span class="va">padj</span> <span class="op">&lt;=</span> <span class="fl">0.05</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 1993</code></pre>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">conditionGenes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">condRes</span><span class="op">)</span><span class="op">[</span><span class="va">condRes</span><span class="op">$</span><span class="va">padj</span> <span class="op">&lt;=</span> <span class="fl">0.05</span><span class="op">]</span>
<span class="va">conditionGenes</span> <span class="op">&lt;-</span> <span class="va">conditionGenes</span><span class="op">[</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">conditionGenes</span><span class="op">)</span><span class="op">]</span></code></pre></div>
</div>
<div id="visualize-most-and-least-significant-gene" class="section level2">
<h2 class="hasAnchor">
<a href="#visualize-most-and-least-significant-gene" class="anchor"></a>Visualize most and least significant gene</h2>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">RColorBrewer</span><span class="op">)</span>
<span class="va">scales</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/RColorBrewer/man/ColorBrewer.html">brewer.pal</a></span><span class="op">(</span><span class="fl">3</span>, <span class="st">"Accent"</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span>

<span class="co"># plot genes</span>
<span class="va">oo</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/order.html">order</a></span><span class="op">(</span><span class="va">condRes</span><span class="op">$</span><span class="va">waldStat</span>, decreasing <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>

<span class="co"># most significant gene</span>
<span class="va">p6</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/tradeSeq/man/plotSmoothers.html">plotSmoothers</a></span><span class="op">(</span><span class="va">tgfb</span>, <span class="fu">assays</span><span class="op">(</span><span class="va">tgfb</span><span class="op">)</span><span class="op">$</span><span class="va">counts</span>,
              gene <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="fu">assays</span><span class="op">(</span><span class="va">tgfb</span><span class="op">)</span><span class="op">$</span><span class="va">counts</span><span class="op">)</span><span class="op">[</span><span class="va">oo</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>,
              alpha <span class="op">=</span> <span class="fl">1</span>, border <span class="op">=</span> <span class="cn">TRUE</span>, curvesCols <span class="op">=</span> <span class="va">scales</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">scale_color_manual</span><span class="op">(</span>values <span class="op">=</span> <span class="va">scales</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">ggtitle</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="fu">assays</span><span class="op">(</span><span class="va">tgfb</span><span class="op">)</span><span class="op">$</span><span class="va">counts</span><span class="op">)</span><span class="op">[</span><span class="va">oo</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<pre><code>## Scale for 'colour' is already present. Adding another scale for 'colour',
## which will replace the existing scale.</code></pre>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Second most significant gene</span>
<span class="va">p7</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/tradeSeq/man/plotSmoothers.html">plotSmoothers</a></span><span class="op">(</span><span class="va">tgfb</span>, <span class="fu">assays</span><span class="op">(</span><span class="va">tgfb</span><span class="op">)</span><span class="op">$</span><span class="va">counts</span>,
              gene <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="fu">assays</span><span class="op">(</span><span class="va">tgfb</span><span class="op">)</span><span class="op">$</span><span class="va">counts</span><span class="op">)</span><span class="op">[</span><span class="va">oo</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span>,
              alpha <span class="op">=</span> <span class="fl">1</span>, border <span class="op">=</span> <span class="cn">TRUE</span>, curvesCols <span class="op">=</span> <span class="va">scales</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">scale_color_manual</span><span class="op">(</span>values <span class="op">=</span> <span class="va">scales</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">ggtitle</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="fu">assays</span><span class="op">(</span><span class="va">tgfb</span><span class="op">)</span><span class="op">$</span><span class="va">counts</span><span class="op">)</span><span class="op">[</span><span class="va">oo</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<pre><code>## Scale for 'colour' is already present. Adding another scale for 'colour',
## which will replace the existing scale.</code></pre>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># least significant gene</span>
<span class="va">p8</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/tradeSeq/man/plotSmoothers.html">plotSmoothers</a></span><span class="op">(</span><span class="va">tgfb</span>, <span class="fu">assays</span><span class="op">(</span><span class="va">tgfb</span><span class="op">)</span><span class="op">$</span><span class="va">counts</span>,
              gene <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="fu">assays</span><span class="op">(</span><span class="va">tgfb</span><span class="op">)</span><span class="op">$</span><span class="va">counts</span><span class="op">)</span><span class="op">[</span><span class="va">oo</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="va">tgfb</span><span class="op">)</span><span class="op">]</span><span class="op">]</span>,
              alpha <span class="op">=</span> <span class="fl">1</span>, border <span class="op">=</span> <span class="cn">TRUE</span>, curvesCols <span class="op">=</span> <span class="va">scales</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">scale_color_manual</span><span class="op">(</span>values <span class="op">=</span> <span class="va">scales</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">ggtitle</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="fu">assays</span><span class="op">(</span><span class="va">tgfb</span><span class="op">)</span><span class="op">$</span><span class="va">counts</span><span class="op">)</span><span class="op">[</span><span class="va">oo</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="va">tgfb</span><span class="op">)</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<pre><code>## Scale for 'colour' is already present. Adding another scale for 'colour',
## which will replace the existing scale.</code></pre>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p6</span></code></pre></div>
<p><img src="TGFB_files/figure-html/unnamed-chunk-10-1.png" width="672"></p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p7</span></code></pre></div>
<p><img src="TGFB_files/figure-html/unnamed-chunk-10-2.png" width="672"></p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p8</span></code></pre></div>
<p><img src="TGFB_files/figure-html/unnamed-chunk-10-3.png" width="672"></p>
</div>
<div id="heatmaps-of-genes-de-between-conditions" class="section level2">
<h2 class="hasAnchor">
<a href="#heatmaps-of-genes-de-between-conditions" class="anchor"></a>Heatmaps of genes DE between conditions</h2>
<p>Below we show heatmaps of the genes DE between conditions. The DE genes in the heatmaps are ordered according to a hierarchical clustering on the TGF-Beta condition.</p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://wilkelab.org/cowplot/">cowplot</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://scales.r-lib.org">scales</a></span><span class="op">)</span>
<span class="co">### based on mean smoother</span>
<span class="va">yhatSmooth</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/tradeSeq/man/predictSmooth.html">predictSmooth</a></span><span class="op">(</span><span class="va">tgfb</span>, gene <span class="op">=</span> <span class="va">conditionGenes</span>, nPoints <span class="op">=</span> <span class="fl">50</span>, tidy <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/r/base/Log.html">log1p</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">yhatSmoothScaled</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/t.html">t</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/apply.html">apply</a></span><span class="op">(</span><span class="va">yhatSmooth</span>,<span class="fl">1</span>, <span class="fu">scales</span><span class="fu">::</span><span class="va"><a href="https://scales.r-lib.org//reference/rescale.html">rescale</a></span><span class="op">)</span><span class="op">)</span>
<span class="va">heatSmooth_TGF</span> <span class="op">&lt;-</span> <span class="fu">pheatmap</span><span class="op">(</span><span class="va">yhatSmoothScaled</span><span class="op">[</span>, <span class="fl">51</span><span class="op">:</span><span class="fl">100</span><span class="op">]</span>,
  cluster_cols <span class="op">=</span> <span class="cn">FALSE</span>,
  show_rownames <span class="op">=</span> <span class="cn">FALSE</span>, show_colnames <span class="op">=</span> <span class="cn">FALSE</span>, main <span class="op">=</span> <span class="st">"TGF-Beta"</span>, legend <span class="op">=</span> <span class="cn">FALSE</span>,
  silent <span class="op">=</span> <span class="cn">TRUE</span>
<span class="op">)</span>

<span class="va">matchingHeatmap_mock</span> <span class="op">&lt;-</span> <span class="fu">pheatmap</span><span class="op">(</span><span class="va">yhatSmoothScaled</span><span class="op">[</span><span class="va">heatSmooth_TGF</span><span class="op">$</span><span class="va">tree_row</span><span class="op">$</span><span class="va">order</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">50</span><span class="op">]</span>,
  cluster_cols <span class="op">=</span> <span class="cn">FALSE</span>, cluster_rows <span class="op">=</span> <span class="cn">FALSE</span>,
  show_rownames <span class="op">=</span> <span class="cn">FALSE</span>, show_colnames <span class="op">=</span> <span class="cn">FALSE</span>, main <span class="op">=</span> <span class="st">"Mock"</span>,
  legend <span class="op">=</span> <span class="cn">FALSE</span>, silent <span class="op">=</span> <span class="cn">TRUE</span> 
<span class="op">)</span>

<span class="va">p9</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://wilkelab.org/cowplot/reference/plot_grid.html">plot_grid</a></span><span class="op">(</span><span class="va">heatSmooth_TGF</span><span class="op">[[</span><span class="fl">4</span><span class="op">]</span><span class="op">]</span>, <span class="va">matchingHeatmap_mock</span><span class="op">[[</span><span class="fl">4</span><span class="op">]</span><span class="op">]</span>, ncol <span class="op">=</span> <span class="fl">2</span><span class="op">)</span>
<span class="va">p9</span></code></pre></div>
<p><img src="TGFB_files/figure-html/unnamed-chunk-11-1.png" width="672"></p>
</div>
<div id="gene-set-enrichment-analysis" class="section level2">
<h2 class="hasAnchor">
<a href="#gene-set-enrichment-analysis" class="anchor"></a>Gene set enrichment analysis</h2>
<p>This is done using the <em>fgsea</em> package <span class="citation">(Korotkevich, Sukhov, and Sergushichev 2016)</span>.</p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">geneSets</span> <span class="op">&lt;-</span> <span class="fu">msigdbr</span><span class="op">(</span>species <span class="op">=</span> <span class="st">"Mus musculus"</span>, category <span class="op">=</span> <span class="st">"C5"</span>, subcategory <span class="op">=</span> <span class="st">"BP"</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">mutate</span><span class="op">(</span>gene_symbol <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/chartr.html">toupper</a></span><span class="op">(</span><span class="va">gene_symbol</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">gene_symbol</span> <span class="op">%in%</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">tgfb</span><span class="op">)</span><span class="op">)</span> 
<span class="va">m_list</span> <span class="op">&lt;-</span> <span class="va">geneSets</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/split.html">split</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">.</span><span class="op">$</span><span class="va">gene_symbol</span>, f <span class="op">=</span> <span class="va">.</span><span class="op">$</span><span class="va">gs_name</span><span class="op">)</span>
<span class="va">statsCond</span> <span class="op">&lt;-</span> <span class="va">condRes</span><span class="op">$</span><span class="va">waldStat</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">statsCond</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">condRes</span><span class="op">)</span>
<span class="va">eaRes</span> <span class="op">&lt;-</span> <span class="fu">fgsea</span><span class="op">(</span>pathways <span class="op">=</span> <span class="va">m_list</span>, stats <span class="op">=</span> <span class="va">statsCond</span>, nperm <span class="op">=</span> <span class="fl">5e4</span>, minSize <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></code></pre></div>
<pre><code>## Warning in fgsea(pathways = m_list, stats = statsCond, nperm = 50000,
## minSize = 10): You are trying to run fgseaSimple. It is recommended to use
## fgseaMultilevel. To run fgseaMultilevel, you need to remove the nperm argument
## in the fgsea function call.</code></pre>
<pre><code>## Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (0.03% of the list).
## The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">eaRes</span> <span class="op">&lt;-</span> <span class="va">eaRes</span> <span class="op">%&gt;%</span> <span class="fu">arrange</span><span class="op">(</span><span class="va">pval</span><span class="op">)</span>
<span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html">kable</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">eaRes</span><span class="op">[</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span>, n <span class="op">=</span> <span class="fl">20</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<table class="table">
<colgroup>
<col width="75%">
<col width="12%">
<col width="12%">
</colgroup>
<thead><tr class="header">
<th align="left">pathway</th>
<th align="right">pval</th>
<th align="right">padj</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">GO_BIOLOGICAL_ADHESION</td>
<td align="right">0.0000200</td>
<td align="right">0.0792784</td>
</tr>
<tr class="even">
<td align="left">GO_ANATOMICAL_STRUCTURE_FORMATION_INVOLVED_IN_MORPHOGENESIS</td>
<td align="right">0.0000600</td>
<td align="right">0.1057046</td>
</tr>
<tr class="odd">
<td align="left">GO_CELL_MOTILITY</td>
<td align="right">0.0000800</td>
<td align="right">0.1057046</td>
</tr>
<tr class="even">
<td align="left">GO_REGULATION_OF_CELL_ADHESION</td>
<td align="right">0.0002000</td>
<td align="right">0.1981960</td>
</tr>
<tr class="odd">
<td align="left">GO_REGULATION_OF_MITOCHONDRIAL_GENE_EXPRESSION</td>
<td align="right">0.0003800</td>
<td align="right">0.3012580</td>
</tr>
<tr class="even">
<td align="left">GO_TRANSCRIPTION_BY_RNA_POLYMERASE_III</td>
<td align="right">0.0005400</td>
<td align="right">0.3567529</td>
</tr>
<tr class="odd">
<td align="left">GO_EXTRACELLULAR_STRUCTURE_ORGANIZATION</td>
<td align="right">0.0008000</td>
<td align="right">0.3567529</td>
</tr>
<tr class="even">
<td align="left">GO_VASCULATURE_DEVELOPMENT</td>
<td align="right">0.0008800</td>
<td align="right">0.3567529</td>
</tr>
<tr class="odd">
<td align="left">GO_CIRCULATORY_SYSTEM_DEVELOPMENT</td>
<td align="right">0.0010000</td>
<td align="right">0.3567529</td>
</tr>
<tr class="even">
<td align="left">GO_CELL_MORPHOGENESIS_INVOLVED_IN_DIFFERENTIATION</td>
<td align="right">0.0010600</td>
<td align="right">0.3567529</td>
</tr>
<tr class="odd">
<td align="left">GO_MITOCHONDRIAL_RNA_METABOLIC_PROCESS</td>
<td align="right">0.0011200</td>
<td align="right">0.3567529</td>
</tr>
<tr class="even">
<td align="left">GO_REGULATION_OF_CELL_POPULATION_PROLIFERATION</td>
<td align="right">0.0013800</td>
<td align="right">0.3567529</td>
</tr>
<tr class="odd">
<td align="left">GO_REGULATION_OF_CELLULAR_COMPONENT_MOVEMENT</td>
<td align="right">0.0014200</td>
<td align="right">0.3567529</td>
</tr>
<tr class="even">
<td align="left">GO_MOTOR_NEURON_AXON_GUIDANCE</td>
<td align="right">0.0014201</td>
<td align="right">0.3567529</td>
</tr>
<tr class="odd">
<td align="left">GO_N_TERMINAL_PROTEIN_AMINO_ACID_MODIFICATION</td>
<td align="right">0.0015000</td>
<td align="right">0.3567529</td>
</tr>
<tr class="even">
<td align="left">GO_TAXIS</td>
<td align="right">0.0015200</td>
<td align="right">0.3567529</td>
</tr>
<tr class="odd">
<td align="left">GO_CELL_CELL_ADHESION</td>
<td align="right">0.0015400</td>
<td align="right">0.3567529</td>
</tr>
<tr class="even">
<td align="left">GO_CELL_MORPHOGENESIS</td>
<td align="right">0.0016200</td>
<td align="right">0.3567529</td>
</tr>
<tr class="odd">
<td align="left">GO_TUBE_DEVELOPMENT</td>
<td align="right">0.0018000</td>
<td align="right">0.3755293</td>
</tr>
<tr class="even">
<td align="left">GO_CELL_PROJECTION_ORGANIZATION</td>
<td align="right">0.0021200</td>
<td align="right">0.4043199</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="session-info" class="section level1">
<h1 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session info</h1>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## R version 4.0.3 (2020-10-10)
## Platform: x86_64-apple-darwin17.0 (64-bit)
## Running under: macOS Big Sur 10.16
## 
## Matrix products: default
## BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
## LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] parallel  stats4    stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
##  [1] scales_1.1.1                cowplot_1.1.1              
##  [3] RColorBrewer_1.1-2          openxlsx_4.2.3             
##  [5] msigdbr_7.2.1               fgsea_1.16.0               
##  [7] pheatmap_1.0.12             condiments_0.1.0           
##  [9] ggplot2_3.3.3               SeuratObject_4.0.0         
## [11] Seurat_4.0.0                slingshot_1.8.0            
## [13] princurve_2.1.6             SingleCellExperiment_1.12.0
## [15] SummarizedExperiment_1.20.0 Biobase_2.50.0             
## [17] GenomicRanges_1.42.0        GenomeInfoDb_1.26.2        
## [19] IRanges_2.24.1              S4Vectors_0.28.1           
## [21] BiocGenerics_0.36.0         MatrixGenerics_1.2.1       
## [23] matrixStats_0.58.0          tradeSeq_1.5.07            
## [25] dplyr_1.0.4                
## 
## loaded via a namespace (and not attached):
##   [1] utf8_1.1.4             reticulate_1.18        tidyselect_1.1.0      
##   [4] htmlwidgets_1.5.3      grid_4.0.3             combinat_0.0-8        
##   [7] docopt_0.7.1           BiocParallel_1.24.1    Rtsne_0.15            
##  [10] pROC_1.17.0.1          munsell_0.5.0          codetools_0.2-18      
##  [13] ragg_1.1.1             ica_1.0-2              future_1.21.0         
##  [16] miniUI_0.1.1.1         withr_2.4.1            colorspace_2.0-0      
##  [19] fastICA_1.2-2          highr_0.8              knitr_1.31            
##  [22] rstudioapi_0.13        ROCR_1.0-11            tensor_1.5            
##  [25] listenv_0.8.0          labeling_0.4.2         slam_0.1-48           
##  [28] GenomeInfoDbData_1.2.4 polyclip_1.10-0        farver_2.0.3          
##  [31] rprojroot_2.0.2        parallelly_1.23.0      vctrs_0.3.6           
##  [34] generics_0.1.0         ipred_0.9-9            xfun_0.21             
##  [37] Ecume_0.9.0            R6_2.5.0               VGAM_1.1-5            
##  [40] locfit_1.5-9.4         bitops_1.0-6           spatstat.utils_2.0-0  
##  [43] cachem_1.0.4           DelayedArray_0.16.1    assertthat_0.2.1      
##  [46] promises_1.2.0.1       nnet_7.3-15            gtable_0.3.0          
##  [49] globals_0.14.0         goftest_1.2-2          timeDate_3043.102     
##  [52] rlang_0.4.10           systemfonts_1.0.1      splines_4.0.3         
##  [55] lazyeval_0.2.2         ModelMetrics_1.2.2.2   yaml_2.2.1            
##  [58] reshape2_1.4.4         abind_1.4-5            httpuv_1.5.5          
##  [61] caret_6.0-86           tools_4.0.3            lava_1.6.8.1          
##  [64] ellipsis_0.3.1         jquerylib_0.1.3        ggridges_0.5.3        
##  [67] Rcpp_1.0.6             plyr_1.8.6             zlibbioc_1.36.0       
##  [70] purrr_0.3.4            RCurl_1.98-1.2         densityClust_0.3      
##  [73] rpart_4.1-15           deldir_0.2-10          pbapply_1.4-3         
##  [76] viridis_0.5.1          zoo_1.8-8              ggrepel_0.9.1         
##  [79] cluster_2.1.1          fs_1.5.0               magrittr_2.0.1        
##  [82] data.table_1.14.0      scattermore_0.7        lmtest_0.9-38         
##  [85] RANN_2.6.1             fitdistrplus_1.1-3     patchwork_1.1.1       
##  [88] mime_0.10              evaluate_0.14          xtable_1.8-4          
##  [91] sparsesvd_0.2          gridExtra_2.3          transport_0.12-2      
##  [94] HSMMSingleCell_1.10.0  compiler_4.0.3         tibble_3.1.0          
##  [97] KernSmooth_2.23-18     crayon_1.4.1           htmltools_0.5.1.1     
## [100] mgcv_1.8-34            later_1.1.0.1          tidyr_1.1.2           
## [103] lubridate_1.7.9.2      DBI_1.1.1              MASS_7.3-53.1         
## [106] Matrix_1.3-2           cli_2.3.1              gower_0.2.2           
## [109] igraph_1.2.6           pkgconfig_2.0.3        pkgdown_1.6.1         
## [112] plotly_4.9.3           recipes_0.1.15         foreach_1.5.1         
## [115] bslib_0.2.4            XVector_0.30.0         prodlim_2019.11.13    
## [118] stringr_1.4.0          digest_0.6.27          sctransform_0.3.2     
## [121] RcppAnnoy_0.0.18       DDRTree_0.1.5          spatstat.data_2.0-0   
## [124] fastmatch_1.1-0        rmarkdown_2.7          leiden_0.3.7          
## [127] uwot_0.1.10            edgeR_3.32.1           kernlab_0.9-29        
## [130] shiny_1.6.0            lifecycle_1.0.0        monocle_2.18.0        
## [133] nlme_3.1-152           jsonlite_1.7.2         desc_1.2.0            
## [136] viridisLite_0.3.0      limma_3.46.0           fansi_0.4.2           
## [139] pillar_1.5.0           lattice_0.20-41        fastmap_1.1.0         
## [142] httr_1.4.2             survival_3.2-7         glue_1.4.2            
## [145] zip_2.1.1              qlcMatrix_0.9.7        FNN_1.1.3             
## [148] spatstat_1.64-1        png_0.1-7              iterators_1.0.13      
## [151] class_7.3-18           stringi_1.5.3          sass_0.3.1            
## [154] textshaping_0.3.1      memoise_2.0.0          e1071_1.7-4           
## [157] irlba_2.3.3            future.apply_1.7.0     ape_5.4-1</code></pre>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references hanging-indent">
<div id="ref-Becht2019">
<p>Becht, Etienne, Leland McInnes, John Healy, Charles-Antoine Dutertre, Immanuel W H Kwok, Lai Guan Ng, Florent Ginhoux, and Evan W Newell. 2019. “Dimensionality reduction for visualizing single-cell data using UMAP.” <em>Nature Biotechnology</em> 37 (1): 38–44. <a href="https://doi.org/10.1038/nbt.4314">https://doi.org/10.1038/nbt.4314</a>.</p>
</div>
<div id="ref-Korotkevich2016">
<p>Korotkevich, Gennady, Vladimir Sukhov, and Alexey Sergushichev. 2016. “Fast gene set enrichment analysis.” <em>bioRxiv</em>, February, 060012. <a href="https://doi.org/10.1101/060012">https://doi.org/10.1101/060012</a>.</p>
</div>
<div id="ref-McFaline-Figueroa2019">
<p>McFaline-Figueroa, José L., Andrew J. Hill, Xiaojie Qiu, Dana Jackson, Jay Shendure, and Cole Trapnell. 2019. “A pooled single-cell genetic screen identifies regulatory checkpoints in the continuum of the epithelial-to-mesenchymal transition.” <em>Nature Genetics</em> 51 (9): 1389–98. <a href="https://doi.org/10.1038/s41588-019-0489-5">https://doi.org/10.1038/s41588-019-0489-5</a>.</p>
</div>
<div id="ref-McInnes2018">
<p>McInnes, Leland, John Healy, and James Melville. 2018. “UMAP: Uniform Manifold Approximation and Projection for Dimension Reduction.” <em>Arxiv</em>, February. <a href="http://arxiv.org/abs/1802.03426">http://arxiv.org/abs/1802.03426</a>.</p>
</div>
<div id="ref-Street2018a">
<p>Street, Kelly, Davide Risso, Russell B. Fletcher, Diya Das, John Ngai, Nir Yosef, Elizabeth Purdom, and Sandrine Dudoit. 2018. “Slingshot: cell lineage and pseudotime inference for single-cell transcriptomics.” <em>BMC Genomics</em> 19 (1): 477. <a href="https://doi.org/10.1186/s12864-018-4772-0">https://doi.org/10.1186/s12864-018-4772-0</a>.</p>
</div>
<div id="ref-Stuart2019">
<p>Stuart, Tim, Andrew Butler, Paul Hoffman, Christoph Hafemeister, Efthymia Papalexi, William M Mauck, Yuhan Hao, Marlon Stoeckius, Peter Smibert, and Rahul Satija. 2019. “Comprehensive Integration of Single-Cell Data.” <em>Cell</em> 177 (7): 1888–1902.e21. <a href="https://doi.org/10.1016/j.cell.2019.05.031">https://doi.org/10.1016/j.cell.2019.05.031</a>.</p>
</div>
<div id="ref-VandenBerge2020">
<p>Van den Berge, Koen, Hector Roux de Bézieux, Kelly Street, Wouter Saelens, Robrecht Cannoodt, Yvan Saeys, Sandrine Dudoit, and Lieven Clement. 2020. “Trajectory-based differential expression analysis for single-cell sequencing data.” <em>Nature Communications</em> 11 (1): 1201. <a href="https://doi.org/10.1038/s41467-020-14766-3">https://doi.org/10.1038/s41467-020-14766-3</a>.</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Hector Roux de Bezieux, Koen Van den Berge, Kelly Street.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
